<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />




<title></title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/readable.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script>
  (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
  (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
  m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
  })(window,document,'script','//www.google-analytics.com/analytics.js','ga');

  ga('create', 'UA-52691601-2', 'auto');
  ga('send', 'pageview');

</script>

<!---

<style type="text/css">

/* padding for bootstrap navbar */
body {
  padding-top: 50px;
  padding-bottom: 40px;
  padding-right: 50px;
  padding-left: -50px;
}
@media (max-width: 979px) {
  body {
    padding-top: 0;
  }
}

 offset scroll position for anchor links (for fixed navbar)  
@media (min-width: 980px) {
  .section h2 {
    padding-top: 52px;
    margin-top: -52px;
  }
  .section h3 {
    padding-top: 52px;
    margin-top: -52px;
  }
}


/* don't use link color in navbar */
.dropdown-menu>li>a {
  color: black;
}

</style>
-->



<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 50px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h2 {
  padding-top: 55px;
  margin-top: -55px;
}
.section h3 {
  padding-top: 55px;
  margin-top: -55px;
}
/* don't use link color in navbar */
.dropdown-menu>li>a {
  color: black;
}
/* some padding for disqus */
#disqus_thread {
  margin-top: 45px;
}
</style>

<link rel="stylesheet" href="libs/font-awesome-4.1.0/css/font-awesome.min.css"/>




<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>

<link rel="stylesheet" href="styles.css" type="text/css" />

</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 66px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 71px;
  margin-top: -71px;
}

.section h2 {
  padding-top: 71px;
  margin-top: -71px;
}
.section h3 {
  padding-top: 71px;
  margin-top: -71px;
}
.section h4 {
  padding-top: 71px;
  margin-top: -71px;
}
.section h5 {
  padding-top: 71px;
  margin-top: -71px;
}
.section h6 {
  padding-top: 71px;
  margin-top: -71px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->






<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Population genetics and genomics in R</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="TOC.html">Table of contents</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Part I
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="Introduction.html">Introduction</a>
    </li>
    <li>
      <a href="Getting_ready_to_use_R.html">Getting ready to use R</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Part II
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="Data_Preparation.html">Data preparation</a>
    </li>
    <li>
      <a href="First_Steps.html">First steps</a>
    </li>
    <li>
      <a href="Population_Strata.html">Population strata and clone correction</a>
    </li>
    <li>
      <a href="Locus_Stats.html">Locus-based statistics and missing data</a>
    </li>
    <li>
      <a href="Genotypic_EvenRichDiv.html">Genotypic evenness, richness, and diversity</a>
    </li>
    <li>
      <a href="Linkage_disequilibrium.html">Linkage disequilibrium</a>
    </li>
    <li>
      <a href="Pop_Structure.html">Population structure</a>
    </li>
    <li>
      <a href="Minimum_Spanning_Networks.html">Minimum Spanning Networks</a>
    </li>
    <li>
      <a href="AMOVA.html">AMOVA</a>
    </li>
    <li>
      <a href="DAPC.html">Discriminant analysis of principal components (DAPC)</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Part III
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="intro_vcf.html">Population genomics and HTS</a>
    </li>
    <li>
      <a href="reading_vcf.html">Reading VCF data</a>
    </li>
    <li>
      <a href="qc.html">Quality control</a>
    </li>
    <li>
      <a href="analysis_of_genome.html">Analysis of genomic data</a>
    </li>
    <li>
      <a href="gbs_analysis.html">Analysis of GBS data</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Workshop
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="workshop.html">Workshop</a>
    </li>
    <li>
      <a href="intro_vcf.html">Introduction</a>
    </li>
    <li>
      <a href="reading_vcf.html">VCF data</a>
    </li>
    <li>
      <a href="gbs_analysis.html">Analysis of GBS data</a>
    </li>
    <li>
      <a href="analysis_of_genome.html">Analysis of genome data</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    About
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="Authors.html">Authors</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Appendices
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="intro_to_R.html">Introduction to R</a>
    </li>
    <li>
      <a href="Data_sets.html">Data sets</a>
    </li>
    <li>
      <a href="funpendix.html">Function glossary</a>
    </li>
    <li>
      <a href="https://github.com/grunwaldlab/Population_Genetics_in_R/">Source Code</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <!-- HOME -->
      <a class="navbar-brand" href="index.html"><b>Population Genetics in R</b></a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <!-- Table of contents -->
        <li class="dropdown">
          <!---<a href="chapters" class="dropdown-toggle" data-toggle="dropdown">Chapters<b class="caret"></b></a>-->
          <a href="chapters" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">Chapters <span class="caret"></span></a>
            <ol class="dropdown-menu">
               <li><a href="Preface.html">i. Preface</a></li>
               <li><a href="Introduction.html">1. Introduction</a></li>
               <li><a href="Getting_ready_to_use_R.html">2. Getting ready to use R</a></li>
               <li><a href="Data_Preparation.html">3. Data preparation</a></li>
               <li><a href="First_Steps.html">4. First steps</a></li>
               <li><a href="Population_Strata.html">5. Population strata and clone correction</a></li>
               <li><a href="Locus_Stats.html">6. Locus-based statistics and missing data</a></li>
               <li><a href="Genotypic_EvenRichDiv.html">7. Genotypic evenness, richness, and diversity</a></li>
               <li><a href="Linkage_disequilibrium.html">8. Linkage disequilibrium</a></li>
               <li><a href="Pop_Structure.html">9. Population structure</a></li>
               <li><a href="Minimum_Spanning_Networks.html">10. Minimum Spanning Networks</a></li>
               <li><a href="AMOVA.html">11. AMOVA</a></li>
               <li><a href="DAPC.html">12. Discriminant analysis of principal components (DAPC)</a></li>
            </ol>
        </li>
        <!-- Appendices dropdown -->
        <li class="dropdown">
          <!---<a href="appendices" class="dropdown-toggle" data-toggle="dropdown">Appendices<b class="caret"></b></a>-->
          <a href="appendices" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">Appendices <span class="caret"></span></a>
          <ul class = "dropdown-menu", role = "menu">
            <li><a href="Data_sets.html">A1. Data sets</a></li>
            <li><a href="funpendix.html">A2. Function glossary</a></li>
            <li><a href="Intro_to_R.html">A3. Introduction to R</a></li>
          </ul>
        </li>
        <!-- Author button -->
        <li class="dropdown">
        	<a href="about" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">About<span class="caret"></span></a>
          <ul class = "dropdown-menu", role = "menu">
        	<li><a href="Authors.html">Authors</a></li>
        	<li>
        		<a href="https://github.com/grunwaldlab/Population_Genetics_in_R", target = "blank">
		    	Source code
		    	</a>
		    </li>
		   </ul>
        </li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->
<!-- 

old rmarkdown version: 

devtools::install_github("rstudio/rmarkdown", ref = "b96214b9ac86b437067a0aa21442203f52face83")


 -->
<!-- <div class="container">
<div class="row">
<div class="span10 offset1"> -->

<div class="fluid-row" id="header">




</div>


<script>
!function(d,s,id){var js,fjs=d.getElementsByTagName(s)[0],p=/^http:/.test(d.location)?'http':'https';if(!d.getElementById(id)){js=d.createElement(s);js.id=id;js.src=p+'://platform.twitter.com/widgets.js';fjs.parentNode.insertBefore(js,fjs);}}(document, 'script', 'twitter-wjs');
</script>
<!-- 

Note that the following HTML cannot be indented because markdown will attempt to
render it as a code block otherwise.

-->
<div class="container">
<h1>
About the authors
</h1>
<!-- 
NIK'S PROFILE
 -->
<div class="row">
<div class="col-md-2">
<div class="imgAbt">
<p><img style="margin-bottom:25px;width=220px;height=220px" src="images/IMG_3734.jpg"  /></p>
</div>
</div>
<div class="col-md-6">
<small> <strong>Niklaus J. Grünwald</strong> is a plant pathologist with the USDA Agricultural Research Service and a Professor in the Department of Botany and Pant Pathology and the Center for Genome Research and Biocomputing at Oregon State University. He received his PhD from UC Davis and conducted postdoctoral research at Cornell University. His research focuses on understanding the evolutionary patterns and processes of emerging pathogens using population genetic and evolutionary approaches and development of computational tools for characterizing pathogens and communities. Members of his group have developed several widely used R packages including: <a href="http://github.com/grunwaldlab/poppr"><em>poppr</em></a>, <a href="https://github.com/grunwaldlab/vcfR"><em>vcfr</em></a>, and <a href="https://github.com/grunwaldlab/metacoder"><em>metacoder</em></a>.
<p>
<p>
<a href="http://grunwaldlab.cgrb.oregonstate.edu/">Grünwald lab</a> | <a href="https://github.com/grunwaldlab">Github</a> | <a href="https://scholar.google.com/citations?user=t55Dt5sAAAAJ&hl=en">Google scholar</a> | <a href="http://www.researcherid.com/rid/K-6041-2013">ResearcherID</a> | <a href="http://orcid.org/0000-0003-1656-7602">Orcid</a> | <a href="https://twitter.com/PhytophthoraLab" class="twitter-follow-button" data-show-count="false"> Follow @PhytophthoraLab </a>
</p>
</small>
</div>
</div>
<!-- 
ZHIAN'S PROFILE
 -->
<div class="row">
<div class="col-md-2">
<div class="imgAbt">
<p><img style="margin-bottom:25px;width=220px;height=220px" src="images/ZK.jpg"  /></p>
</div>
</div>
<div class="col-md-6">
<small> <strong>Zhian N. Kamvar</strong> is a postdoctroal researcher in the Plant Pathology Department at the University of Nebraska-Lincoln. He is the lead developer of the <a href="http://github.com/grunwaldlab/poppr"><em>poppr</em></a> R package and a contributor to other packages such as <a href="http://github.com/thibautjombart/adegenet"><em>adegenet</em></a> and <a href="http://github.com/dwinter/mmod"><em>mmod</em></a>.
<p>
<p>
<a href="https://scholar.google.com/citations?user=UPGtLSkAAAAJ&hl=en">Google scholar</a> | <a href="https://www.researchgate.net/profile/Zhian_Kamvar">ResearchGate</a> | <a href="http://orcid.org/0000-0003-1458-7108">Orcid</a> | <a href="https://twitter.com/ZKamvar" class="twitter-follow-button" data-show-count="false"> Follow @ZKamvar </a> |
</p>
</small>
</div>
</div>
<!-- 
SYDNEY'S PROFILE
 -->
<div class="row">
<div class="col-md-2">
<div class="imgAbt">
<p><img style="margin-bottom:25px;width=220px;height=220px" src="images/SE.jpg"  /></p>
</div>
</div>
<div class="col-md-6">
<small> <strong>Sydney E. Everhart</strong> is an Assistant Professor at the University of Nebraska- Lincoln. Her research utilizes population genetics and genomics to better understand disease dynamics and evolution of fungal plant pathogens. She developed the micrsoatellite markers and data used in the <a href="Data_sets.html#microsatellite-haploid-clonal-monpop"><code>monpop</code></a> data set and currently works with genomes, GBS data, and SSR data for <em>Phytophthora ramorum</em>.
<p>
<p>
<a href="http://everhart.unl.edu/">Everhart lab</a> | <a href="https://twitter.com/SydneyEverhart" class="twitter-follow-button" data-show-count="false"> Follow @SydneyEverhart </a>
</p>
</small>
</div>
</div>
<!-- 
BRIAN'S PROFILE
 -->
<div class="row">
<div class="col-md-2">
<div class="imgAbt">
<p><img style="margin-bottom:25px;width=220px;height=220px" src="images/BJK.jpg"  /></p>
</div>
</div>
<div class="col-md-6">
<small> <strong>Brian J. Knaus</strong> is an postdoctoral researcher with the USDA Agricultural Research Service. He is the lead developer of the <a href="https://github.com/grunwaldlab/vcfR"><em>vcfr</em></a> package. Brian’s current research uses genomics approaches to survey the plant pathogens <em>Phytophthora infestans</em> (late blight of potato and tomato), <em>Podosphaera macularis</em> (powdery mildew of hop), and <em>Pseudoperonospora humuli</em> (downey mildew of hop) for population structure and to identify genomic features associated with phenotypes responsible for mating type determination and fungicide resistance.
<p>
<p>
<a href="http://brianknaus.com/research/">Brian’s webpage</a> <a href="https://twitter.com/knaus_brian" class="twitter-follow-button" data-show-count="false"> | Follow @Brian J. Kanus | <a href="http://orcid.org/0000-0003-1665-4343">Orcid</a> | <a href="https://scholar.google.com/citations?user=19wcuSwAAAAJ&hl=en">Google scholar</a> </a>
</p>
</small>
</div>
</div>
<!-- 
JAVIER'S PROFILE
 -->
<div class="row">
<div class="col-md-2">
<div class="imgAbt">
<p><img style="margin-bottom:25px;width=220px;height=220px" src="images/JFT.jpg"  /></p>
</div>
</div>
<div class="col-md-6">
<small> <strong>Javier F. Tabima</strong> is a graduate student in the Department of Botany and Plant Pathology at Oregon State University. His main research interests are to study the molecular processes of host speciation using genomics. His dissertation is based in contructing a bridge between evolutionary biology and plant pathology. Javier has contributed to the <a href="http://github.com/grunwaldlab/poppr"><em>poppr</em></a> R package, and has created a set of packages for computational plant pathology such as <a href="https://github.com/grunwaldlab/Microbe-ID">Microbe-ID</a> and other tools in the current process of development.
<p>
<p>
<a href="http://people.oregonstate.edu/~tabimaj/">Javier’s webpage</a>| <a href="https://scholar.google.com/citations?user=t1P5G4QAAAAJ&hl=en">Google scholar</a> | <a href="https://www.researchgate.net/profile/Javier_Tabima">ResearchGate</a> | <a href="http://orcid.org/0000-0002-3603-2691">Orcid</a> | <a href="https://twitter.com/tabima" class="twitter-follow-button" data-show-count="false"> Follow @Tabima </a> |
</p>
</small>
</div>
</div>
</div>

<!-- </div>
</div>
</div>
 -->



</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
