<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta http-equiv="Content-Style-Type" content="text/css" />
<meta name="generator" content="pandoc" />



<title>About the authors</title>

<script src="libs/jquery-1.11.0/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1.0" />
<link href="libs/bootstrap-2.3.2/css/readable.min.css" rel="stylesheet" />
<link href="libs/bootstrap-2.3.2/css/bootstrap-responsive.min.css" rel="stylesheet" />
<script src="libs/bootstrap-2.3.2/js/bootstrap.min.js"></script>
<script>
  (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
  (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
  m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
  })(window,document,'script','//www.google-analytics.com/analytics.js','ga');

  ga('create', 'UA-52691601-2', 'auto');
  ga('send', 'pageview');

</script>

<style type="text/css">

/* padding for bootstrap navbar */
body {
  padding-top: 50px;
  padding-bottom: 40px;
  padding-right: 50px;
  padding-left: -50px;
}
@media (max-width: 979px) {
  body {
    padding-top: 0;
  }
}

/* offset scroll position for anchor links (for fixed navbar)  */
@media (min-width: 980px) {
  .section h2 {
    padding-top: 52px;
    margin-top: -52px;
  }
  .section h3 {
    padding-top: 52px;
    margin-top: -52px;
  }
}


/* don't use link color in navbar */
.dropdown-menu>li>a {
  color: black;
}

</style>

<style type="text/css">code{white-space: pre;}</style>
<link rel="stylesheet"
      href="libs/highlight/default.css"
      type="text/css" />
<script src="libs/highlight/highlight.js"></script>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs && document.readyState && document.readyState === "complete") {
   window.setTimeout(function() {
      hljs.initHighlighting();
   }, 0);
}
</script>



</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
</style>
<div class="container-fluid main-container">


<div class="navbar navbar-default navbar-fixed-top">
  <div class="navbar-inner">
    <div class="container">
      <button type="button" class="btn btn-navbar" data-toggle="collapse" data-target=".nav-collapse">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="brand" href="index.html"><b>Population Genetics in R</b></a>
      <div class="nav-collapse collapse">
        <ul class="nav">
          <li class="dropdown">
            <a href="chapters" class="dropdown-toggle" data-toggle="dropdown">Chapters<b class="caret"></b></a>
            <ul class="dropdown-menu">
               <li><a href="Preface.html">i. Preface</a></li>
               <li><a href="C01_Introduction.html">1. Introduction</a></li>
               <li><a href="C02_Getting_ready_to_use_R.html">2. Getting ready to use R</a></li>
               <li><a href="C03_Data_Preparation.html">3. Data Preparation</a></li>
               <li><a href="C04_First_Steps.html">4. First steps</a></li>
               <li><a href="C05_Population_Hierarchies.html">5. Population hierarchies</a></li>
               <li><a href="C06_Locus_Stats.html">6. Locus-based statistics and missing data</a></li>
               <li><a href="C07_Genotypic_EvenRichDiv.html">7. Genotypic evenness, richness, and diversity</a></li>
               <li><a href="C08_Linkage_disequilibrium.html">8. Linkage disequilibrium</a></li>
               <li><a href="C09_Pop_Structure.html">9. Population structure</a></li>
               <li><a href="C10_AMOVA.html">10. AMOVA</a></li>
               <li><a href="C11_DAPC.html">11. Discriminant analysis of principal components (DAPC)</a></li>
               </ul>
            </li>
          <li class="dropdown">
            <a href="appendices" class="dropdown-toggle" data-toggle="dropdown">Appendices<b class="caret"></b></a>
            <ul class = "dropdown-menu">
               <li><a href="A1_Data_sets.html">A1. Data sets</a></li>
               <li><a href="A2_Function_Glossary.html">A2. Function glossary</a></li>
            </ul>
          </li>
          
        <ul class="nav">
          <li><a href="Authors.html">About</a></li>
          <li class="dropdown">

        </ul>
      </div><!--/.nav-collapse -->
    </div>
  </div>
</div>

<div class="container">
<div class="row">
<div class="span10 offset1">

<div id="header">
<h1 class="title">About the authors</h1>
</div>


<TABLE Frame=void cellpadding="5" cellspacing = "5">
 <TR>
   <TD VALIGN=TOP>
<div style="width: 150px">
<img src="images/IMG_3734.jpg" height="400px" width="300px"/>
</div>
</TD>
   <TD VALIGN=TOP>
<small><strong>Niklaus J. Grünwald</strong> is a plant pathologist with the USDA Agricultural Research Service and a Professor in the Department of Botany and Pant Pathology and the Center for Genome Research and Biocomputing at Oregon State University. He received his PhD from UC Davis and conducted postdoctoral research at Cornell University. His research focuses on understanding the evolutionary patterns and processes of emerging pathogens using population genetic and evolutionary approaches.
<p>
<a href="http://grunwaldlab.cgrb.oregonstate.edu/">Grünwald lab</a> | <a href="http://www.researcherid.com/rid/K-6041-2013">ResearcherID</a> | <a href="http://orcid.org/0000-0003-1656-7602">Orcid</a> | <a href="https://www.researchgate.net/profile/Niklaus_Grunwald/?ev=hdr_xprf">ResearchGate</a> | <a href="http://impactstory.org/NiklausGrunwald">ImpactStory</a> | <a href="https://twitter.com/PhytophthoraLab" class="twitter-follow-button" data-show-count="false">Follow <span class="citation">(<span class="citeproc-not-found" data-reference-id="PhytophthoraLab"><strong>???</strong></span>)</span></a>
<script>!function(d,s,id){var js,fjs=d.getElementsByTagName(s)[0],p=/^http:/.test(d.location)?'http':'https';if(!d.getElementById(id)){js=d.createElement(s);js.id=id;js.src=p+'://platform.twitter.com/widgets.js';fjs.parentNode.insertBefore(js,fjs);}}(document, 'script', 'twitter-wjs');</script></p>
</small>
</TD>
 </TR>
  <TR>
   <TD VALIGN=TOP>
<div style="width: 150px">
<img src="images/ZK.jpg" height="400px" width="300px"/>
</div>
</TD>
   <TD VALIGN=TOP>
<small><strong>Zhian N. Kamvar</strong> is a graduate student in the Department of Botany and Pant Pathology at Oregon State University. His research is focused on genome sequencing and population genomics of the nursery pathogen <em>Phytophthora syringae</em>. He is the lead developer of the <a href="http://github.com/grunwaldlab/poppr"><em>poppr</em> R package</a>. Zhian is also co-host of the award-winning radio program <a href="http://poweredbyorange.com/2013/12/17/science-through-a-microphone/">Inspiration Dissemination</a>. </small>
</TD>
 </TR>
 <TR>
   <TD VALIGN=TOP>
<div style="width: 150px">
<img src="images/SydneyEverhart.JPG" height="400px" width="300px"/>
</div>
</TD>
   <TD VALIGN=TOP>
<small><strong>Sydney E. Everhart</strong> is a USDA-AFRI-NIFA Postdoctoral Fellow at Oregon State University. Her research utilizes population genetics and genomics to better understand disease dynamics and evolution of fungal plant pathogens. She developed the micrsoatellite markers and data used in the <a href="A1_Data_sets.html#microsatellite-haploid-clonal-monpop"><code>monpop</code></a> data set and currently works with genomes, GBS data, and SSR data for <em>Phytophthora ramorum</em> in Nik’s lab. After the APS meeting, she will be starting as an Assistant Professor at the University of Nebraska-Lincoln studying the emergence and evolution of fungicide resistance within fungal plant pathogen populations. She is currently seeking a graduate student interested in population genetics of fungi. </small>
</TD>
 </TR>
</TABLE>


<div class="references">

</div>

</div>
</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
$(document).ready(function () {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
});

</script>

<!-- dynamically load mathjax for compatibility with --self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "libs/mathjax-2.3.0/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
