<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta http-equiv="Content-Style-Type" content="text/css" />
<meta name="generator" content="pandoc" />



<title>10: Discriminant analysis of principal components (DAPC)</title>

<script src="libs/jquery-1.11.0/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1.0" />
<link href="libs/bootstrap-2.3.2/css/readable.min.css" rel="stylesheet" />
<link href="libs/bootstrap-2.3.2/css/bootstrap-responsive.min.css" rel="stylesheet" />
<script src="libs/bootstrap-2.3.2/js/bootstrap.min.js"></script>
<style type="text/css">

/* padding for bootstrap navbar */
body {
  padding-top: 50px;
  padding-bottom: 40px;
}
@media (max-width: 979px) {
  body {
    padding-top: 0;
  }
}

/* offset scroll position for anchor links (for fixed navbar)  */
@media (min-width: 980px) {
  .section h2 {
    padding-top: 52px;
    margin-top: -52px;
  }
  .section h3 {
    padding-top: 52px;
    margin-top: -52px;
  }
}


/* don't use link color in navbar */
.dropdown-menu>li>a {
  color: black;
}

</style>

<style type="text/css">code{white-space: pre;}</style>
<link rel="stylesheet"
      href="libs/highlight/default.css"
      type="text/css" />
<script src="libs/highlight/highlight.js"></script>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs && document.readyState && document.readyState === "complete") {
   window.setTimeout(function() {
      hljs.initHighlighting();
   }, 0);
}
</script>



</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
</style>
<div class="container-fluid main-container">


<div class="navbar navbar-default navbar-fixed-top">
  <div class="navbar-inner">
    <div class="container">
      <button type="button" class="btn btn-navbar" data-toggle="collapse" data-target=".nav-collapse">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="brand" href="index.html"><b>Population Genetics in R</b></a>
      <div class="nav-collapse collapse">
        <ul class="nav">
          <li><a href="TOC.html">Home</a></li>
          <li class="dropdown">
            <a href="chapters" class="dropdown-toggle" data-toggle="dropdown">Chapters<b class="caret"></b></a>
            <ul class="dropdown-menu">
               <li><a href="Foreword.html">i. Preface</a></li>
               <li><a href="C1_Introduction.html">1. Introduction</a></li>
               <li><a href="C2_Getting_ready_to_use_R.html">2. Getting ready to use R</a></li>
               <li><a href="C3_Data_Preparation.html">3. Data formatting</a></li>
               <li><a href="C3_Data_format_loading.html">4. First steps</a></li>
               <li><a href="C4_Population_Hierarchies.html">5. Population hierarchies</a></li>
               <li><a href="C5_Locus_Stats.html">6. Locus-based statistics and missing data</a></li>
               <li><a href="C6_Genotypic_EvenRichDiv.html">7. Genotypic evenness, richness, and diversity</a></li>
               <li><a href="C7_Linkage_disequilibrium.html">8. Linkage disequilibrium</a></li>
               <li><a href="C8_Pop_Structure.html">9. Population structure</a></li>
               <li><a href="C9_AMOVA.html">10. AMOVA</a></li>
               <li><a href="C10_DAPC.html">11. Discriminant analysis of principal components (DAPC)</a></li>
               </ul>
            </li>
          <li class="dropdown">
            <a href="appendices" class="dropdown-toggle" data-toggle="dropdown">Appendices<b class="caret"></b></a>
            <ul class = "dropdown-menu">
               <li><a href="A1_Data_sets.html">A1. Data sets</a></li>
               <li><a href="A2_Function_Glossary.html">A2. Function glossary</a></li>
            </ul>
          </li>
          
        <ul class="nav">
          <li><a href="Authors.html">About</a></li>
          <li class="dropdown">

        </ul>
      </div><!--/.nav-collapse -->
    </div>
  </div>
</div>

<div class="container">
<div class="row">
<div class="span10 offset1">

<div id="header">
<h1 class="title">10: Discriminant analysis of principal components (DAPC)</h1>
</div>

<div id="TOC">
<ul>
<li><a href="#dapc-analysis-of-the-h3n2-influenza-strains">DAPC analysis of the H3N2 influenza strains</a></li>
<li><a href="#conclusions">Conclusions</a></li>
<li><a href="#references">References</a></li>
</ul>
</div>

<p>Often we want to infer population structure by determining the number of clusters (groups) observed without prior knowledge. Several approaches can be used to infer groups such as for example K-means clustering, Bayesian clustering using STRUCTURE, and multivariate methods such as Discriminant Analysis of Principal Components (DAPC) (<a href="http://www.biomedcentral.com/1471-2156/11/94">Jombart et al. 2010</a>; <a href="http://www.genetics.org/content/155/2/945.abstract">Pritchard et al. 2000</a>; <a href="http://www.annualreviews.org/doi/abs/10.1146/annurev-phyto-072910-095246?journalCode=phyto">Grünwald &amp; Goss, 2011</a>). A STRUCTURE-like approach assumes that markers are not linked and that populations are panmictic (<a href="http://www.genetics.org/content/155/2/945.abstract">Pritchard et al. 2000</a>). To use model-free methods K-means clustering based on genetic distance or DAPC are more convenient approaches for populations that are clonal or partially clonal. Here we explore DAPC further.</p>
<div id="dapc-analysis-of-the-h3n2-influenza-strains" class="section level2">
<h2>DAPC analysis of the H3N2 influenza strains</h2>
<p>DAPC was pioneered by Jombart and colleagues (<a href="http://www.biomedcentral.com/1471-2156/11/94">Jombart et al. 2010</a>) and can be used to infer the number of clusters of genetically related individuals. In this multivariate statistical approach variance in the sample is partitioned into a between-group and within-group component, in an effort to maximize discrimination between groups. In DAPC, data is first transformed using a principal components analysis (PCA) and subsequently clusters are identified using discriminant analysis (DA). This tutorial is based on the <a href="http://adegenet.r-forge.r-project.org/files/tutorial-dapc.pdf">vignette</a> written by Thibaut Jombart. We encourage the user to explore this vignette further. The vignette can also be opened within R by executing <code>adegenetTutorial(&quot;dapc&quot;)</code>.</p>
<p>We will use the seasonal influenza dataset H3N2 data containing 1903 isolates genotyped for 125 SNPs located in the hemagglutinin segment. This dataset as well as the <code>dapc()</code> function is part of the <a href="http://adegenet.r-forge.r-project.org"><em>adegenet</em></a> package.</p>
<pre class="r"><code># DAPC requires the adegenet package. Let&#39;s load this package:
library(adegenet)
data(H3N2)      # load the H3N2 inflluenza data. Tye ?H3N2 for more info.
pop(H3N2) &lt;- factor(H3N2$other$epid)
dapc.H3N2 &lt;- dapc(H3N2, var.contrib=TRUE, scale=FALSE, n.pca=30, n.da=10)
scatter(dapc.H3N2, cell=0, pch=18:23, cstar=0, mstree=TRUE, lwd=2, lty=2)</code></pre>
<p><img src="C10_DAPC_files/figure-html/unnamed-chunk-2.png" title="plot of chunk unnamed-chunk-2" alt="plot of chunk unnamed-chunk-2" width="960" /></p>
<p>The <code>dapc()</code> variables refer to:</p>
<ul>
<li>the dataset H3N2</li>
<li><code>var.contrib</code> when set to FALSE, flag removes contribution from original variables from output. We will look at this later to see which loci are responsible for separatig populations.</li>
<li><code>n.pca</code> is the number of axes retained in the Principal Component Analysis (PCA). If set to NULL selection is interactive, which we avoid here but can be explored by the user.</li>
<li><code>n.da</code> is the number of axes retained in the Discriminant Analysis (DA). If set to NULL selection is interactive, which we avoid here but can be explored by the user.</li>
</ul>
<p>The <code>scatter()</code> function is part of the <em>ade4</em> pacakge and plots results of a DAPC analysis.</p>
<p>As you can see, each year between 2001 to 2005 is a cluster of H3N2 strains separated by axis 1. In contrast, axis 2 separates the strains observed in the 2006 cluster from the clusters observed during 2001-5, indicating that the strains observed in 2006 are genetically distinct.</p>
<p>Next, let’s assess if there are alleles that most differentiate the 2006 cluster from those in other years.</p>
<pre class="r"><code>set.seed(4)
contrib &lt;- loadingplot(dapc.H3N2$var.contr, axis=2, thres=.07, lab.jitter=1)</code></pre>
<p><img src="C10_DAPC_files/figure-html/unnamed-chunk-3.png" title="plot of chunk unnamed-chunk-3" alt="plot of chunk unnamed-chunk-3" width="960" /></p>
<p>It looks like SNPs at position 384 and 906 are involved. Let’s check this further by looking at allele frequencies by year:</p>
<pre class="r"><code>temp &lt;- seploc(H3N2)
snp906 &lt;- truenames(temp[[&quot;906&quot;]])$tab #truenames{adegenet} returns elements of a genind object
snp399 &lt;- truenames(temp[[&quot;399&quot;]])$tab
freq906 &lt;- apply(snp906, 2, function(e) tapply(e, pop(H3N2), mean, na.rm=TRUE))
freq399 &lt;- apply(snp399, 2, function(e) tapply(e, pop(H3N2), mean, na.rm=TRUE))
freq906</code></pre>
<pre><code>##         906.c  906.t
## 2001 0.000000 1.0000
## 2002 0.000000 1.0000
## 2003 0.000000 1.0000
## 2004 0.000000 1.0000
## 2005 0.002155 0.9978
## 2006 0.616071 0.3839</code></pre>
<pre class="r"><code>freq399</code></pre>
<pre><code>##         399.c  399.t
## 2001 0.000000 1.0000
## 2002 0.000000 1.0000
## 2003 0.000000 1.0000
## 2004 0.001848 0.9982
## 2005 0.002079 0.9979
## 2006 0.357143 0.6429</code></pre>
<p>Note that a new allele appeared in 2005 for SNP locus 906 and 2004 for locus 399 separating populations along axis 2.</p>
<pre class="r"><code>par(mfrow=c(1,2), mar=c(5.1,4.1,4.1,.1),las=3)
matplot(freq906, pch=c(&quot;c&quot;,&quot;t&quot;), type=&quot;b&quot;,xlab=&quot;year&quot;,ylab=&quot;allele frequency&quot;, xaxt=&quot;n&quot;, cex=1.5, main=&quot;SNP # 906&quot;)
axis(side=1, at=1:6, lab=2001:2006)
matplot(freq399, pch=c(&quot;c&quot;,&quot;t&quot;), type=&quot;b&quot;, xlab=&quot;year&quot;,ylab=&quot;allele frequency&quot;, xaxt=&quot;n&quot;, cex=1.5, main=&quot;SNP #399&quot;)
axis(side=1, at=1:6, lab=2001:2006)</code></pre>
<p><img src="C10_DAPC_files/figure-html/unnamed-chunk-5.png" title="plot of chunk unnamed-chunk-5" alt="plot of chunk unnamed-chunk-5" width="960" /></p>
<p>This plot nicely illustrates the effect of mutation, followed by selection or drift in the seasonal H3N2 influenza virus.</p>
</div>
<div id="conclusions" class="section level2">
<h2>Conclusions</h2>
<p>DAPC is a wonderful tool for exploring structure of populations based on PCA and DA without making assumptions of panmixia. Thus, this technique provides a robust alternative to Bayesian clustering methods like STRUCTURE (<a href="http://www.genetics.org/content/155/2/945.abstract">Pritchard et al. 2000</a>) that should not be used for clonal or partially clonal populations.</p>
<p>DAPC analysis is inherently interactive and cannot be scripted <em>a priori</em>. Please refer to the <a href="http://adegenet.r-forge.r-project.org/files/tutorial-dapc.pdf">vignette</a> written by Thibaut Jombart for a more interactive analysis.</p>
</div>
<div id="references" class="section level2">
<h2>References</h2>
<p> 
NJ Grünwald, EM Goss, (2011) Evolution and population genetics of exotic and re-emerging pathogens: Novel tools and approaches. <em>Annual Review of Phytopathology</em> <strong>49</strong> 249-267 <a href="http://www.annualreviews.org/doi/abs/10.1146/annurev-phyto-072910-095246?journalCode=phyto">http://www.annualreviews.org/doi/abs/10.1146/annurev-phyto-072910-095246?journalCode=phyto</a>
</p><p> 
T Jombart, S Devillard, F Balloux, (2010) Discriminant analysis of principal components: a new method for the analysis of genetically structured populations. <em>BMC genetics</em> <strong>11</strong> (1) 94-NA <a href="http://www.biomedcentral.com/1471-2156/11/94">http://www.biomedcentral.com/1471-2156/11/94</a>
</p><p> 
JK Pritchard, M Stephens, P. Donnelly, (2000) Inference of population structure using multilocus genotype data. <em>Genetics</em> <strong>155</strong> (2) 945-959 <a href="http://www.genetics.org/content/155/2/945.abstract">http://www.genetics.org/content/155/2/945.abstract</a>
</p>


</div>

</div>
</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
$(document).ready(function () {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
});

</script>

<!-- dynamically load mathjax for compatibility with --self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "libs/mathjax-2.3.0/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
