<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta http-equiv="Content-Style-Type" content="text/css" />
<meta name="generator" content="pandoc" />



<title>Chapter 4: First Analysis</title>

<script src="libs/jquery-1.11.0/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1.0" />
<link href="libs/bootstrap-2.3.2/css/readable.min.css" rel="stylesheet" />
<link href="libs/bootstrap-2.3.2/css/bootstrap-responsive.min.css" rel="stylesheet" />
<script src="libs/bootstrap-2.3.2/js/bootstrap.min.js"></script>
<style type="text/css">

/* padding for bootstrap navbar */
body {
  padding-top: 50px;
  padding-bottom: 40px;
}
@media (max-width: 979px) {
  body {
    padding-top: 0;
  }
}

/* offset scroll position for anchor links (for fixed navbar)  */
@media (min-width: 980px) {
  .section h2 {
    padding-top: 52px;
    margin-top: -52px;
  }
  .section h3 {
    padding-top: 52px;
    margin-top: -52px;
  }
}


/* don't use link color in navbar */
.dropdown-menu>li>a {
  color: black;
}

</style>

<style type="text/css">code{white-space: pre;}</style>
<link rel="stylesheet"
      href="libs/highlight/default.css"
      type="text/css" />
<script src="libs/highlight/highlight.js"></script>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs && document.readyState && document.readyState === "complete") {
   window.setTimeout(function() {
      hljs.initHighlighting();
   }, 0);
}
</script>



</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
</style>
<div class="container-fluid main-container">


<div class="navbar navbar-default navbar-fixed-top">
  <div class="navbar-inner">
    <div class="container">
      <button type="button" class="btn btn-navbar" data-toggle="collapse" data-target=".nav-collapse">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="brand" href="index.html"><b>Population Genetics in R</b></a>
      <div class="nav-collapse collapse">
        <ul class="nav">
          <li><a href="TOC.html">Home</a></li>
          <li class="dropdown">
            <a href="chapters" class="dropdown-toggle" data-toggle="dropdown">Chapters<b class="caret"></b></a>
            <ul class="dropdown-menu">
               <li><a href="Preface.html">i. Preface</a></li>
               <li><a href="C1_Introduction.html">1. Introduction</a></li>
               <li><a href="C2_Getting_ready_to_use_R.html">2. Getting ready to use R</a></li>
               <li><a href="C3_Data_Preparation.html">3. Data formatting</a></li>
               <li><a href="C4_First_Steps.html">4. First steps</a></li>
               <li><a href="C5_Population_Hierarchies.html">5. Population hierarchies</a></li>
               <li><a href="C6_Locus_Stats.html">6. Locus-based statistics and missing data</a></li>
               <li><a href="C7_Genotypic_EvenRichDiv.html">7. Genotypic evenness, richness, and diversity</a></li>
               <li><a href="C8_Linkage_disequilibrium.html">8. Linkage disequilibrium</a></li>
               <li><a href="C9_Pop_Structure.html">9. Population structure</a></li>
               <li><a href="C10_AMOVA.html">10. AMOVA</a></li>
               <li><a href="C11_DAPC.html">11. Discriminant analysis of principal components (DAPC)</a></li>
               </ul>
            </li>
          <li class="dropdown">
            <a href="appendices" class="dropdown-toggle" data-toggle="dropdown">Appendices<b class="caret"></b></a>
            <ul class = "dropdown-menu">
               <li><a href="A1_Data_sets.html">A1. Data sets</a></li>
               <li><a href="A2_Function_Glossary.html">A2. Function glossary</a></li>
            </ul>
          </li>
          
        <ul class="nav">
          <li><a href="Authors.html">About</a></li>
          <li class="dropdown">

        </ul>
      </div><!--/.nav-collapse -->
    </div>
  </div>
</div>

<div class="container">
<div class="row">
<div class="span10 offset1">

<div id="header">
<h1 class="title">Chapter 4: First Analysis</h1>
</div>

<div id="TOC">
<ul>
<li><a href="#analyzing-the-genotype-accumulation-curve">Analyzing the genotype accumulation curve</a></li>
<li><a href="#allele-frequencies-missing-data-and-ploidy">Allele frequencies, missing data, and ploidy</a></li>
<li><a href="#calculating-genotypic-diversity">Calculating genotypic diversity</a></li>
<li><a href="#references">References</a></li>
</ul>
</div>

<p>This chapter introduces basic use of and navigation in <em>poppr</em> as well as good practices when starting an analysis. We take a glance at the genotypic diversity observed in populations and allelic frequencies observed in your loci by population. Both of these are good first steps to eliminate data entry errors, check for missing/rare data, and make sure all loci conform to expectations given your analyses.</p>
<p>For these examples, we will be using a data set that is already built into <em>poppr</em> containing 86 individuals of <em>Phytophthora infestans</em> genotyped over 11 microsatellite loci (<a href="http://grunwaldlab.cgrb.oregonstate.edu/irish-potato-famine-pathogen-phytophthora-infestans-originated-central-mexico-rather-andes">Goss et al. 2014</a>)</p>
<div id="analyzing-the-genotype-accumulation-curve" class="section level2">
<h2>Analyzing the genotype accumulation curve</h2>
<p>A genotype accumulation curve is a tool that allows you to assess how much power you have to discriminate between unique individuals given a random sample of <span class="math">\(n\)</span> loci. This analysis is particularly important for clonal organisms to confirm that a plateau has been reached in the number of loci necessary to discriminate individuals. Below, we will analyze the <code>Pinf</code> data set.</p>
<pre class="r"><code>data(Pinf) # Load the data
Pinf       # We expect a maximum of 72 Multilocus Genotypes</code></pre>
<pre><code>## 
## This is a genclone object
## -------------------------
## Genotype information:
## 
##    72 multilocus genotypes
##    86 tetraploid individuals
##    11 codominant loci
## 
## Population information:
## 
##     2 hierarchical levels - Continent Country
##     2 populations defined - South America North America</code></pre>
<pre class="r"><code>gac &lt;- genotype_curve(Pinf, sample = 1000, quiet = TRUE)</code></pre>
<p><img src="C4_First_Steps_files/figure-html/unnamed-chunk-2.png" title="Genotype accumulation curve" alt="Genotype accumulation curve" width="700px" /></p>
<p>We specified <code>sample = 1000</code> in our function call. This means that for each boxplot, <span class="math">\(n\)</span> loci were randomly sampled 1000 times in order to create the distribution. Since this data has been curated, we can see that we have reached the plateau with 11 loci. Try seeing what happens when you use a data set of sexual populations such as <code>microbov</code> or <code>nancycats</code>. Also, what happens when you use dominant AFLP data such as <code>Aeut</code>?</p>
</div>
<div id="allele-frequencies-missing-data-and-ploidy" class="section level2">
<h2>Allele frequencies, missing data, and ploidy</h2>
<p>A good first step after loading data is to look for missing data, rare alleles and overall quality of your data:</p>
<pre class="r"><code>data(Pinf)
locus_table(Pinf)</code></pre>
<pre><code>## 
##  allele = Number of observed alleles
##  1-D = Simpson index
##  Hexp = Nei&#39;s 1978 expected heterozygosity
## ------------------------------------------</code></pre>
<pre><code>##       summary
## locus  allele    1-D   Hexp Evenness
##   Pi02 10.000  0.633  0.703    0.663
##   D13  25.000  0.884  0.921    0.587
##   Pi33  2.000  0.012  0.023    0.322
##   Pi04  4.000  0.578  0.771    0.785
##   Pi4B  7.000  0.669  0.780    0.707
##   Pi16  6.000  0.403  0.484    0.507
##   G11  21.000  0.839  0.881    0.544
##   Pi56  3.000  0.361  0.541    0.707
##   Pi63  3.000  0.413  0.619    0.641
##   Pi70  3.000  0.279  0.419    0.580
##   Pi89 11.000  0.615  0.677    0.578
##   mean  8.636  0.517  0.620    0.602</code></pre>
<pre class="r"><code>info_table(Pinf, type = &quot;missing&quot;, plot = TRUE)</code></pre>
<p><img src="C4_First_Steps_files/figure-html/unnamed-chunk-3.png" title="Missing data plot" alt="Missing data plot" width="700px" /></p>
<pre><code>##                Locus
## Population      Pi02   D13 Pi33 Pi04 Pi4B  Pi16 G11 Pi56  Pi63 Pi70  Pi89
##   South America    .     .    .    .    .     .   .    .     .    .     .
##   North America    . 0.104    .    .    . 0.062   .    . 0.021    . 0.021
##   Total            . 0.058    .    .    . 0.035   .    . 0.012    . 0.012
##                Locus
## Population       Mean
##   South America     .
##   North America 0.019
##   Total         0.011</code></pre>
<p>If your data consists of polyploid SSR markers, you would code the unobserved alleles as “0”. The <code>Pinf</code> data set is a perfect example of this. It is a subset of a slightly larger tetraploid data set. In this example, each row represents an isolate and each column represents a locus:</p>
<pre class="r"><code>tail(genind2df(Pinf, sep = &quot;/&quot;))</code></pre>
<pre><code>##                  pop            Pi02             D13            Pi33
## PiPE22 South America 000/000/160/160 000/000/142/144 000/000/203/203
## PiPE23 South America 000/000/160/160 000/000/140/140 000/000/203/203
## PiPE24 South America 000/000/160/160 000/000/142/144 000/000/203/203
## PiPE25 South America 000/000/160/160 000/000/142/144 000/000/203/203
## PiPE26 South America 000/000/162/162 000/000/142/142 000/000/203/203
## PiPE27 South America 000/000/154/160 000/000/136/138 000/000/203/203
##                   Pi04            Pi4B            Pi16             G11
## PiPE22 000/000/166/170 000/000/213/217 000/000/176/178 000/152/162/200
## PiPE23 000/000/166/170 000/000/213/217 000/000/176/178 000/152/160/200
## PiPE24 000/000/166/170 000/000/213/217 000/000/176/178 000/152/162/200
## PiPE25 000/000/166/170 000/000/213/217 000/000/176/178 000/152/162/200
## PiPE26 000/000/166/170 000/000/205/217 000/000/176/178 000/000/156/166
## PiPE27 000/000/166/170 000/000/213/217 000/000/176/178 000/152/170/200
##                   Pi56            Pi63            Pi70            Pi89
## PiPE22 000/000/176/176 000/148/151/157 000/000/189/192 000/000/179/179
## PiPE23 000/000/176/176 000/148/151/157 000/000/189/192 000/000/179/179
## PiPE24 000/000/176/176 000/148/151/157 000/000/189/192 000/000/179/179
## PiPE25 000/000/176/176 000/148/151/157 000/000/189/192 000/000/179/179
## PiPE26 000/000/176/176 000/000/157/157 000/000/192/195 000/000/179/181
## PiPE27 000/000/176/176 000/148/151/157 000/000/189/192 000/000/179/179</code></pre>
<p>To observe the different levels of ploidy in your data, use the function <code>info_table</code> with the argument <code>type = &quot;ploidy&quot;</code>:</p>
<pre class="r"><code>Pinf.ploidy &lt;- info_table(Pinf, type = &quot;ploidy&quot;, plot = TRUE)</code></pre>
<p><img src="C4_First_Steps_files/figure-html/unnamed-chunk-5.png" title="Plot of ploidy" alt="Plot of ploidy" width="700px" /></p>
<pre class="r"><code>tail(Pinf.ploidy)</code></pre>
<pre><code>##         Loci
## Samples  Pi02 D13 Pi33 Pi04 Pi4B Pi16 G11 Pi56 Pi63 Pi70 Pi89
##   PiPE22    2   2    2    2    2    2   3    2    3    2    2
##   PiPE23    2   2    2    2    2    2   3    2    3    2    2
##   PiPE24    2   2    2    2    2    2   3    2    3    2    2
##   PiPE25    2   2    2    2    2    2   3    2    3    2    2
##   PiPE26    2   2    2    2    2    2   2    2    2    2    2
##   PiPE27    2   2    2    2    2    2   3    2    3    2    2</code></pre>
<p>You are now able to assess the quality of your data and examine allelic diversity in your populations. In the next section we will learn how to calculate genotypic diversity.</p>
</div>
<div id="calculating-genotypic-diversity" class="section level2">
<h2>Calculating genotypic diversity</h2>
<p>Let us get a first impression of the diversity found in this data using the summary function, <code>poppr</code>:</p>
<pre class="r"><code>poppr(Pinf)</code></pre>
<pre><code>## | South America 
## | North America 
## | Total</code></pre>
<pre><code>##             Pop  N MLG eMLG    SE    H    G  Hexp   E.5    Ia  rbarD File
## 1 South America 38  29 29.0 0.000 3.27 23.3 0.983 0.883 2.873 0.3446 Pinf
## 2 North America 48  43 34.5 0.989 3.69 34.9 0.992 0.871 0.223 0.0240 Pinf
## 3         Total 86  72 34.6 1.529 4.19 57.8 0.994 0.875 0.652 0.0717 Pinf</code></pre>
<p>We can see statistics printed for each individual and the total population. The fields you see in the output include:</p>
<ul>
<li><code>Pop</code> - Population name.</li>
<li><code>N</code> - Number of individuals observed.</li>
<li><code>MLG</code> - Number of multilocus genotypes (MLG) observed.</li>
<li><code>eMLG</code> - The number of expected MLG at the smallest sample size ≥ 10 based on rarefaction (<a href="http://www.jstor.org/discover/10.2307/1934145?uid=3739856&uid=2&uid=4&uid=3739256&sid=21103760010461">Hurlbert, 1971</a>).</li>
<li><code>SE</code> - Standard error based on eMLG.</li>
<li><code>H</code> - Shannon-Wiener Index of MLG diversity (<a href="http://cm.bell-labs.com/cm/ms/what/shannonday/shannon1948.pdf">Shannon, 2001</a>).</li>
<li><code>G</code> - Stoddart and Taylor’s Index of MLG diversity (<a href="http://www.genetics.org/content/118/4/705">Stoddart &amp; Taylor, 1988</a>).</li>
<li><code>Hexp</code> - Nei’s expected heterozygosity (denoted elsewhere as D) (<a href="http://www.genetics.org/content/89/3/583.abstract">Nei, 1978</a>).</li>
<li><code>E.5</code> - Evenness, E5 (<a href="">Pielou, 1975</a>; <a href="">Ludwig &amp; Reynolds, 1988</a>; <a href="http://apsjournals.apsnet.org/doi/abs/10.1094/PHYTO.2003.93.6.738">Grünwald et al. 2003</a>).</li>
<li><code>Ia</code> - The index of association, <span class="math">\(I_A\)</span> (<a href="http://www.genetics.org/content/96/2/523">Brown et al. 1980</a>; <a href="http://www.pnas.org/content/90/10/4384">Smith et al. 1993</a>).</li>
<li><code>rbarD</code> - The standardized index of association, <span class="math">\(\bar{r}_d\)</span> (<a href="http://onlinelibrary.wiley.com/doi/10.1046/j.1471-8278.2000.00014.x/abstract">Agapow &amp; Burt, 2001</a>).</li>
</ul>
<p>What does all this mean? Both populations have a similar number of individuals (<span class="math">\(N\)</span> = 38 and 48) sampled. However, one population has 29 and the other 43 <span class="math">\(MLG\)</span> while both populations together have combined 86 <span class="math">\(MLG\)</span>. Genotypic diversity (either <span class="math">\(H\)</span> or <span class="math">\(G\)</span>) is higher in population 2 than population 1, while evenness is similar. Let’s ignore <span class="math">\(I_A\)</span> and <span class="math">\(\bar{r}_d\)</span> for now, as these are measures of linkage disequilibrium that will be covered in <a href="C7_Linkage_disequilibrium.html">Chapter 7</a>.</p>
<p>Next, let’s calculate <span class="math">\(MLG\)</span> for each population, e.g. “South America” and “North America” populations:</p>
<pre class="r"><code>P.tab &lt;- mlg.table(Pinf)</code></pre>
<pre><code>## | South America</code></pre>
<p><img src="C4_First_Steps_files/figure-html/unnamed-chunk-71.png" title="Multilocus Genotype Table" alt="Multilocus Genotype Table" width="700px" /></p>
<pre><code>## | North America</code></pre>
<p><img src="C4_First_Steps_files/figure-html/unnamed-chunk-72.png" title="Multilocus Genotype Table" alt="Multilocus Genotype Table" width="700px" /></p>
<p>The figures provide histograms to see how evenly <span class="math">\(MLGs\)</span> are distributed within each population. Both populations have genotypes that occur a few times and only a few genotypes that occur more than 2 times (as expected from the high <span class="math">\(E_5\)</span> discussed above). Try this analysis on other published data sets such as that of the root rot pathogen <em>Aphanomyces euteiches</em>, stored in <em>poppr</em> as <code>Aeut</code>. See if you can produce similar tables and graphs and compare it to the published paper (<a href="http://apsjournals.apsnet.org/doi/abs/10.1094/PHYTO-96-1134">Grünwald &amp; Hoheisel, 2006</a>).</p>
<p>This concludes our first introduction on how to count genotypes and determine genotypic diversity, evenness and richness in populations. In the <a href="C5_Population_Hierarchies">next chapter</a> we will learn how to subset and clone-correct populations.</p>
</div>
<div id="references" class="section level2">
<h2>References</h2>
<p> 
P-M Agapow, A Burt, (2001) Indices of multilocus linkage disequilibrium. <em>Molecular Ecology Notes</em> <strong>1</strong> (1-2) 101-102 <a href="http://onlinelibrary.wiley.com/doi/10.1046/j.1471-8278.2000.00014.x/abstract">http://onlinelibrary.wiley.com/doi/10.1046/j.1471-8278.2000.00014.x/abstract</a>
</p><p> 
AHD Brown, MW Feldman, E Nevo, (1980) Multilocus structure of natural populations of Hordeum spontaneum. <em>Genetics</em> <strong>96</strong> (2) 523-536 <a href="http://www.genetics.org/content/96/2/523">http://www.genetics.org/content/96/2/523</a>
</p><p> 
EM Goss, JF Tabima, DEL Cooke, S Restrepo, WE Fry, GA Forbes, VJ Fieland, M Cardenas, NJ Grünwald, (2014) The Irish famine pathogen Phytophthora infestans originated in central Mexico rather than the Andes. <em>Proceedings of the National Academy of Sciences</em> <strong>111</strong> (24) 8791-8796-NA <a href="http://grunwaldlab.cgrb.oregonstate.edu/irish-potato-famine-pathogen-phytophthora-infestans-originated-central-mexico-rather-andes">http://grunwaldlab.cgrb.oregonstate.edu/irish-potato-famine-pathogen-phytophthora-infestans-originated-central-mexico-rather-andes</a>
</p><p> 
NJ Grünwald, SB Goodwin, MG Milgroom, WE Fry, (2003) Analysis of genotypic diversity data for populations of microorganisms. <em>Phytopathology</em> <strong>93</strong> (6) 738-746 <a href="http://apsjournals.apsnet.org/doi/abs/10.1094/PHYTO.2003.93.6.738">http://apsjournals.apsnet.org/doi/abs/10.1094/PHYTO.2003.93.6.738</a>
</p><p> 
NJ Grünwald, G-A Hoheisel, (2006) Hierarchical analysis of diversity, selfing, and genetic differentiation in populations of the oomycete Aphanomyces euteiches. <em>Phytopathology</em> <strong>96</strong> (10) 1134-1141 <a href="http://apsjournals.apsnet.org/doi/abs/10.1094/PHYTO-96-1134">http://apsjournals.apsnet.org/doi/abs/10.1094/PHYTO-96-1134</a>
</p><p> 
SH Hurlbert, (1971) The nonconcept of species diversity: a critique and alternative parameters. <em>Ecology</em> <strong>52</strong> (4) 577-586 <a href="http://www.jstor.org/discover/10.2307/1934145?uid=3739856&uid=2&uid=4&uid=3739256&sid=21103760010461">http://www.jstor.org/discover/10.2307/1934145?uid=3739856&amp;uid=2&amp;uid=4&amp;uid=3739256&amp;sid=21103760010461</a>
</p><p> 
JA Ludwig, JF Reynolds, (1988) Statistical ecology: a primer in methods and computing.
</p><p> 
M Nei, (1978) Estimation of average heterozygosity and genetic distance from a small number of individuals. <em>Genetics</em> <strong>89</strong> (3) 583-590 <a href="http://www.genetics.org/content/89/3/583.abstract">http://www.genetics.org/content/89/3/583.abstract</a>
</p><p> 
EC Pielou, (1975) Ecological diversity.
</p><p> 
CE Shannon, (2001) A mathematical theory of communication. <em>ACM SIGMOBILE Mobile Computing and Communications Review</em> <strong>5</strong> (1) 3-55 <a href="http://cm.bell-labs.com/cm/ms/what/shannonday/shannon1948.pdf">http://cm.bell-labs.com/cm/ms/what/shannonday/shannon1948.pdf</a>
</p><p> 
JM Smith, NH Smith, M O’Rourke, BG Spratt, (1993) How clonal are bacteria. <em>Proceedings of the National Academy of Sciences</em> <strong>90</strong> (10) 4384-4388 <a href="http://www.pnas.org/content/90/10/4384">http://www.pnas.org/content/90/10/4384</a>
</p><p> 
JA Stoddart, JF Taylor, (1988) Genotypic diversity: estimation and prediction in samples. <em>Genetics</em> <strong>118</strong> (4) 705-711 <a href="http://www.genetics.org/content/118/4/705">http://www.genetics.org/content/118/4/705</a>
</p>


</div>

</div>
</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
$(document).ready(function () {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
});

</script>

<!-- dynamically load mathjax for compatibility with --self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "libs/mathjax-2.3.0/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
