<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta http-equiv="Content-Style-Type" content="text/css" />
<meta name="generator" content="pandoc" />



<title>5: Locus stats, heterozygosity, HWE</title>

<script src="libs/jquery-1.11.0/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1.0" />
<link href="libs/bootstrap-2.3.2/css/readable.min.css" rel="stylesheet" />
<link href="libs/bootstrap-2.3.2/css/bootstrap-responsive.min.css" rel="stylesheet" />
<script src="libs/bootstrap-2.3.2/js/bootstrap.min.js"></script>
<style type="text/css">

/* padding for bootstrap navbar */
body {
  padding-top: 50px;
  padding-bottom: 40px;
}
@media (max-width: 979px) {
  body {
    padding-top: 0;
  }
}

/* offset scroll position for anchor links (for fixed navbar)  */
@media (min-width: 980px) {
  .section h2 {
    padding-top: 52px;
    margin-top: -52px;
  }
  .section h3 {
    padding-top: 52px;
    margin-top: -52px;
  }
}


/* don't use link color in navbar */
.dropdown-menu>li>a {
  color: black;
}

</style>

<style type="text/css">code{white-space: pre;}</style>
<link rel="stylesheet"
      href="libs/highlight/default.css"
      type="text/css" />
<script src="libs/highlight/highlight.js"></script>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs && document.readyState && document.readyState === "complete") {
   window.setTimeout(function() {
      hljs.initHighlighting();
   }, 0);
}
</script>



</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
</style>
<div class="container-fluid main-container">


<div class="navbar navbar-default navbar-fixed-top">
  <div class="navbar-inner">
    <div class="container">
      <button type="button" class="btn btn-navbar" data-toggle="collapse" data-target=".nav-collapse">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="brand" href="index.html"><b>Population Genetics in R</b></a>
      <div class="nav-collapse collapse">
        <ul class="nav">
          <li><a href="TOC.html">Home</a></li>
          <li class="dropdown">
            <a href="chapters" class="dropdown-toggle" data-toggle="dropdown">Chapters<b class="caret"></b></a>
            <ul class="dropdown-menu">
               <li><a href="Foreword.html">i. Preface</a></li>
               <li><a href="C1_Introduction.html">1. Introduction</a></li>
               <li><a href="C2_Getting_ready_to_use_R.html">2. Getting ready to use R</a></li>
               <li><a href="C3_Data_Preparation.html">3. Data formatting</a></li>
               <li><a href="C3_Data_format_loading.html">4. First steps</a></li>
               <li><a href="C4_Population_Hierarchies.html">5. Population hierarchies</a></li>
               <li><a href="C5_Locus_Stats.html">6. Locus-based statistics and missing data</a></li>
               <li><a href="C6_Genotypic_EvenRichDiv.html">7. Genotypic evenness, richness, and diversity</a></li>
               <li><a href="C7_Linkage_disequilibrium.html">8. Linkage disequilibrium</a></li>
               <li><a href="C8_Pop_Structure.html">9. Population structure</a></li>
               <li><a href="C9_AMOVA.html">10. AMOVA</a></li>
               <li><a href="C10_DAPC.html">11. Discriminant analysis of principal components (DAPC)</a></li>
               </ul>
            </li>
          <li class="dropdown">
            <a href="appendices" class="dropdown-toggle" data-toggle="dropdown">Appendices<b class="caret"></b></a>
            <ul class = "dropdown-menu">
               <li><a href="A1_Data_sets.html">A1. Data sets</a></li>
               <li><a href="A2_Function_Glossary.html">A2. Function glossary</a></li>
            </ul>
          </li>
          
        <ul class="nav">
          <li><a href="Authors.html">About</a></li>
          <li class="dropdown">

        </ul>
      </div><!--/.nav-collapse -->
    </div>
  </div>
</div>

<div class="container">
<div class="row">
<div class="span10 offset1">

<div id="header">
<h1 class="title">5: Locus stats, heterozygosity, HWE</h1>
</div>

<div id="TOC">
<ul>
<li><a href="#locus-summary-statistics">Locus summary statistics</a></li>
<li><a href="#phylogenetically-uninformative-loci">Phylogenetically uninformative loci</a></li>
<li><a href="#missing-data">Missing data</a></li>
<li><a href="#removing-loci-and-genotypes">Removing loci and genotypes</a></li>
<li><a href="#replacing-missing-data">Replacing missing data</a></li>
<li><a href="#hardy-weinberg-equilibrium">Hardy-Weinberg equilibrium</a></li>
<li><a href="#references">References</a></li>
</ul>
</div>

<p>A rigorous population genetic analysis looks closely at the data to assess quality and identify outliers or problems in the data such as erroneous allele calls. This chapter focuses on analysis on a per-locus level. While there are statistics that analyze populations across loci, it is important to analyze each locus independently to make sure that one locus is not introducing bias into the analysis.</p>
<blockquote>
<p><strong>Note:</strong> Many of these statistics are specific to co-dominant data.</p>
</blockquote>
<div id="locus-summary-statistics" class="section level2">
<h2>Locus summary statistics</h2>
<p>A quick way to assess quality of the data is to determine the number, diversity, expected heterozygosity, and evenness of the alleles at each locus. As an example, we will use data for the fungal-like protist <em>Phytophthora infestans</em> from (<a href="http://grunwaldlab.cgrb.oregonstate.edu/irish-potato-famine-pathogen-phytophthora-infestans-originated-central-mexico-rather-andes">Goss et al. 2014</a>). First, we’ll use the function <code>locus_table</code> to get all of the statistics mentioned above. For documentation on this function type <code>?locus_table</code>. Here is a first look at each locus:</p>
<pre class="r"><code>library(poppr)     # Make sure poppr is loaded if you haven&#39;t done so already.
data(Pinf)         # P. infestans data set from Mexico and South America
locus_table(Pinf)</code></pre>
<pre><code>## 
##  allele = Number of observed alleles
##  1-D = Simpson index
##  Hexp = Nei&#39;s 1978 expected heterozygosity
## ------------------------------------------</code></pre>
<pre><code>##       summary
## locus  allele    1-D   Hexp Evenness
##   Pi02 10.000  0.633  0.703    0.663
##   D13  25.000  0.884  0.921    0.587
##   Pi33  2.000  0.012  0.023    0.322
##   Pi04  4.000  0.578  0.771    0.785
##   Pi4B  7.000  0.669  0.780    0.707
##   Pi16  6.000  0.403  0.484    0.507
##   G11  21.000  0.839  0.881    0.544
##   Pi56  3.000  0.361  0.541    0.707
##   Pi63  3.000  0.413  0.619    0.641
##   Pi70  3.000  0.279  0.419    0.580
##   Pi89 11.000  0.615  0.677    0.578
##   mean  8.636  0.517  0.620    0.602</code></pre>
<p>We can see here that we have widely variable number of alleles per locus and that we actually have a single locus that only has two alleles, Pi33. This locus also has low diversity, low expected heterozygosity and is very uneven in allele distribution. This is a sign that this might be a phylogenetically uninformative locus, where we have two alleles and one is occurring at a minor frequency. We should explore data analysis with and without this locus. Let’s first see if both of these alleles exist in both populations of this data set.</p>
<pre class="r"><code>locus_table(Pinf, pop = &quot;North America&quot;)</code></pre>
<pre><code>## 
##  allele = Number of observed alleles
##  1-D = Simpson index
##  Hexp = Nei&#39;s 1978 expected heterozygosity
## ------------------------------------------</code></pre>
<pre><code>##       summary
## locus  allele    1-D   Hexp Evenness
##   Pi02  9.000  0.690  0.776    0.653
##   D13  21.000  0.895  0.940    0.684
##   Pi33  2.000  0.021  0.041    0.353
##   Pi04  4.000  0.545  0.727    0.764
##   Pi4B  5.000  0.596  0.745    0.736
##   Pi16  6.000  0.425  0.510    0.498
##   G11  15.000  0.824  0.883    0.625
##   Pi56  3.000  0.335  0.502    0.647
##   Pi63  3.000  0.310  0.465    0.568
##   Pi70  2.000  0.203  0.406    0.595
##   Pi89 11.000  0.627  0.690    0.549
##   mean  7.364  0.497  0.608    0.607</code></pre>
<pre class="r"><code>locus_table(Pinf, pop = &quot;South America&quot;)</code></pre>
<pre><code>## 
##  allele = Number of observed alleles
##  1-D = Simpson index
##  Hexp = Nei&#39;s 1978 expected heterozygosity
## ------------------------------------------</code></pre>
<pre><code>##       summary
## locus  allele   1-D  Hexp Evenness
##   Pi02   5.00  0.54  0.67     0.83
##   D13   13.00  0.83  0.90     0.67
##   Pi33   1.00     .               
##   Pi04   4.00  0.61  0.81     0.81
##   Pi4B   7.00  0.70  0.82     0.78
##   Pi16   3.00  0.35  0.53     0.69
##   G11   14.00  0.80  0.87     0.63
##   Pi56   2.00  0.39  0.78     0.81
##   Pi63   3.00  0.50  0.75     0.73
##   Pi70   3.00  0.37  0.55     0.62
##   Pi89   2.00  0.48  0.97     0.97
##   mean   5.18  0.51  0.76     0.75</code></pre>
</div>
<div id="phylogenetically-uninformative-loci" class="section level2">
<h2>Phylogenetically uninformative loci</h2>
<p>We can see that the South American populations are fixed for this allele, thus it would not be a bad idea to remove that locus from downstream analyses. We can do this using the function <code>informloci</code>. This will remove loci that contain less than a given percentage of divergent individuals (the default is <span class="math">\(2/N\)</span>, where <span class="math">\(N\)</span> equals the number of individuals in the data set).</p>
<pre class="r"><code>nLoc(Pinf)  # Let&#39;s look at our data set, note how many loci we have.</code></pre>
<pre><code>## [1] 11</code></pre>
<pre class="r"><code>iPinf &lt;- informloci(Pinf)</code></pre>
<pre><code>## cutoff value: 2.326 percent ( 2 individuals ).
## 1 uninformative locus found: Pi33</code></pre>
<pre class="r"><code>nLoc(iPinf) # Note that we have 1 less locus</code></pre>
<pre><code>## [1] 10</code></pre>
<p>So, how does this affect multi-locus based statistics? We can see immediately that it didn’t affect the number of multilocus genotypes, let’s take a look at the index of association:</p>
<pre class="r"><code>poppr(Pinf)</code></pre>
<pre><code>## | South America 
## | North America 
## | Total</code></pre>
<pre><code>##             Pop  N MLG eMLG    SE    H    G  Hexp   E.5    Ia  rbarD File
## 1 South America 38  29 29.0 0.000 3.27 23.3 0.983 0.883 2.873 0.3446 Pinf
## 2 North America 48  43 34.5 0.989 3.69 34.9 0.992 0.871 0.223 0.0240 Pinf
## 3         Total 86  72 34.6 1.529 4.19 57.8 0.994 0.875 0.652 0.0717 Pinf</code></pre>
<pre class="r"><code>poppr(iPinf)</code></pre>
<pre><code>## | South America 
## | North America 
## | Total</code></pre>
<pre><code>##             Pop  N MLG eMLG    SE    H    G  Hexp   E.5    Ia  rbarD  File
## 1 South America 38  29 29.0 0.000 3.27 23.3 0.983 0.883 2.873 0.3446 iPinf
## 2 North America 48  43 34.5 0.989 3.69 34.9 0.992 0.871 0.225 0.0255 iPinf
## 3         Total 86  72 34.6 1.529 4.19 57.8 0.994 0.875 0.655 0.0750 iPinf</code></pre>
<p>We can see that it increased ever so slightly for the “North America” and “Total” populations, but not the “South America” population as expected given the fixed alleles at locus P33.</p>
</div>
<div id="missing-data" class="section level2">
<h2>Missing data</h2>
<p>It is often important to asses the percentage of missing data. The <em>poppr</em> function <code>info_table</code> will help you visualize missing data so that you can assess how to treat these further using the function <code>missingno</code>. For this example, we will use the nancycats data set as it contains a wide variety of possabilities for missing data:</p>
<pre class="r"><code>data(nancycats)
info_table(nancycats, plot = TRUE)</code></pre>
<p><img src="C5_Locus_Stats_files/figure-html/missing_table.png" title="Plot of missing data" alt="Plot of missing data" width="700px" /></p>
<pre><code>##           Locus
## Population  fca8 fca23 fca43 fca45 fca77 fca78 fca90 fca96 fca37  Mean
##      1     0.200     .     .     .     .     .     .     .     . 0.022
##      2         .     .     .     .     .     .     .     .     .     .
##      3         .     .     .     .     .     .     .     .     .     .
##      4         .     .     .     .     .     .     .     .     .     .
##      5         .     .     .     .     .     .     .     .     .     .
##      6         .     .     .     .     .     .     .     .     .     .
##      7     0.357     .     .     .     .     .     .     .     . 0.040
##      8         .     .     .     .     .     .     .     .     .     .
##      9         .     .     .     .     .     .     .     .     .     .
##      10        .     .     .     .     .     .     .     .     .     .
##      11    0.150     .     . 0.400     .     .     . 0.050     . 0.067
##      12    0.214     .     .     .     .     .     .     .     . 0.024
##      13        .     .     .     .     .     .     .     .     .     .
##      14    0.412     .     .     .     .     .     .     .     . 0.046
##      15        .     .     .     .     .     .     .     .     .     .
##      16        .     .     .     .     .     .     .     .     .     .
##      17        .     .     . 1.000     .     .     . 0.615     . 0.179
##      Total 0.084     .     . 0.089     .     .     . 0.038     . 0.023</code></pre>
<p>Here we see a few things. The data set has an average of 2.34% missing data overall. More alarming, perhaps is the fact that population 17 has not been genotyped at locus fca45 at all and that locus fca8 shows missing data across many populations. Many analyses in <em>poppr</em> can be performed with missing data in place as it will be either considered an extra allele in the case of MLG calculations or will be interpolated to not contribute to the distance measure used for the index of association. If you want to specifically treat missing data, you can use the function <code>missingno</code> to remove loci or individuals, or replace missing data with zeroes or the average values of the locus.</p>
</div>
<div id="removing-loci-and-genotypes" class="section level2">
<h2>Removing loci and genotypes</h2>
<p>When removing loci or genotypes, you can specify a cutoff representing the percent missing to be removed. The default is <code>0.05</code> (5%).</p>
<pre class="r"><code>nanloc &lt;- missingno(nancycats, &quot;loci&quot;) # Removing loci with greater than 5% missing.</code></pre>
<pre><code>## 
##  Found 617 missing values.
##  2 loci contained missing values greater than 5%.
##  Removing 2 loci : fca8 fca45</code></pre>
<pre class="r"><code>miss &lt;- info_table(nanloc, plot = TRUE, scale = FALSE)</code></pre>
<p><img src="C5_Locus_Stats_files/figure-html/unnamed-chunk-2.png" title="Plot of missing data" alt="Plot of missing data" width="700px" /></p>
<blockquote>
<p><strong>Advanced Users:</strong> when <code>scale = TRUE</code>, the color scale will be set so that the warmest color corresponds to the highest value.</p>
</blockquote>
<p>We only removed two loci. If we wanted to make sure we removed everything, we could set <code>cutoff = 0</code>.</p>
<pre class="r"><code>nanloc &lt;- missingno(nancycats, &quot;loci&quot;, cutoff = 0)</code></pre>
<pre><code>## 
##  Found 617 missing values.
##  3 loci contained missing values greater than 0%.
##  Removing 3 loci : fca8 fca45 fca96</code></pre>
<pre class="r"><code>miss &lt;- info_table(nanloc, plot = TRUE)</code></pre>
<pre><code>## No Missing Data Found!</code></pre>
<p>Again, removing individuals is also relatively easy:</p>
<pre class="r"><code>nanind &lt;- missingno(nancycats, &quot;geno&quot;)</code></pre>
<pre><code>## 
##  Found 617 missing values.
##  38 genotypes contained missing values greater than 5%.
##  Removing 38 genotypes : N215 N216 N188 N189 N190 N191 N192 N298 N299 N300 
## N301 N302 N303 N304 N310 N195 N197 N198 N199 N200 N201 N206 N182 N184 N186 N282 
## N283 N288 N291 N292 N293 N294 N295 N296 N297 N281 N289 N290</code></pre>
<pre class="r"><code>miss &lt;- info_table(nanind, plot = TRUE)</code></pre>
<pre><code>## No Missing Data Found!</code></pre>
<pre class="r"><code>nanind &lt;- missingno(nancycats, &quot;geno&quot;, cutoff = 0)</code></pre>
<pre><code>## 
##  Found 617 missing values.
##  38 genotypes contained missing values greater than 0%.
##  Removing 38 genotypes : N215 N216 N188 N189 N190 N191 N192 N298 N299 N300 
## N301 N302 N303 N304 N310 N195 N197 N198 N199 N200 N201 N206 N182 N184 N186 N282 
## N283 N288 N291 N292 N293 N294 N295 N296 N297 N281 N289 N290</code></pre>
<pre class="r"><code>miss &lt;- info_table(nanind, plot = TRUE) </code></pre>
<pre><code>## No Missing Data Found!</code></pre>
<p><code>Missingno</code> removes individuals based on the percent of missing data relative to the number of loci. Let’s remove all individuals with 2 missing loci:</p>
<pre class="r"><code>nanind &lt;- missingno(nancycats, &quot;geno&quot;, cutoff = 2/nLoc(nancycats))</code></pre>
<pre><code>## 
##  Found 617 missing values.
##  1 genotype contained missing values greater than 22.2222222222222%.
##  Removing 1 genotype : N310</code></pre>
<pre class="r"><code>miss &lt;- info_table(nanind, plot = TRUE) </code></pre>
<p><img src="C5_Locus_Stats_files/figure-html/unnamed-chunk-5.png" title="Plot of missing data" alt="Plot of missing data" width="700px" /></p>
<p>We only found one individual in population 11.</p>
</div>
<div id="replacing-missing-data" class="section level2">
<h2>Replacing missing data</h2>
<p>Replacement of missing data occurs for each allele over all loci. It will replace all missing data in your data set. There are two options: replacement of missing data with zeroes, in fact recoding these as another allelic state, or replacement of missing data with the average allele frequency observed.</p>
<p>Note that the first population in the data set has 20% missing data at the first locus. Here is the un-replaced data for reference:</p>
<pre class="r"><code>nan1 &lt;- popsub(nancycats, 1, drop = TRUE) # Dropping alleles not in that population.
nan1[loc = &quot;L1&quot;]@tab</code></pre>
<pre><code>##    L1.1 L1.2 L1.3 L1.4
## 01   NA   NA   NA   NA
## 02   NA   NA   NA   NA
## 03  0.0  0.5  0.0  0.5
## 04  0.5  0.5  0.0  0.0
## 05  0.5  0.5  0.0  0.0
## 06  0.0  0.5  0.0  0.5
## 07  0.0  1.0  0.0  0.0
## 08  0.0  0.5  0.0  0.5
## 09  0.0  0.0  0.5  0.5
## 10  0.0  1.0  0.0  0.0</code></pre>
<pre class="r"><code>nanzero &lt;- missingno(nancycats, &quot;zero&quot;)</code></pre>
<pre><code>## 
##  Replaced 617 missing values</code></pre>
<pre class="r"><code>popsub(nanzero, 1, drop = TRUE)[loc = &quot;L1&quot;]@tab</code></pre>
<pre><code>##    L1.1 L1.2 L1.3 L1.4
## 01  0.0  0.0  0.0  0.0
## 02  0.0  0.0  0.0  0.0
## 03  0.0  0.5  0.0  0.5
## 04  0.5  0.5  0.0  0.0
## 05  0.5  0.5  0.0  0.0
## 06  0.0  0.5  0.0  0.5
## 07  0.0  1.0  0.0  0.0
## 08  0.0  0.5  0.0  0.5
## 09  0.0  0.0  0.5  0.5
## 10  0.0  1.0  0.0  0.0</code></pre>
<p>The <code>NA</code>s have been replaced with zeroes. Now let’s look at what happens when we replace with <code>&quot;mean&quot;</code>.</p>
<pre class="r"><code>nanmean &lt;- missingno(nancycats, &quot;mean&quot;)</code></pre>
<pre><code>## 
##  Replaced 617 missing values</code></pre>
<pre class="r"><code>popsub(nanmean, 1, drop = TRUE)[loc = &quot;L1&quot;]@tab</code></pre>
<pre><code>##       L1.01    L1.02   L1.03   L1.04    L1.05   L1.06   L1.07   L1.08
## 01 0.002304 0.002304 0.01382 0.06682 0.002304 0.04608 0.05069 0.07604
## 02 0.002304 0.002304 0.01382 0.06682 0.002304 0.04608 0.05069 0.07604
## 03 0.000000 0.000000 0.00000 0.00000 0.000000 0.00000 0.00000 0.00000
## 04 0.000000 0.000000 0.00000 0.00000 0.000000 0.00000 0.00000 0.50000
## 05 0.000000 0.000000 0.00000 0.00000 0.000000 0.00000 0.00000 0.50000
## 06 0.000000 0.000000 0.00000 0.00000 0.000000 0.00000 0.00000 0.00000
## 07 0.000000 0.000000 0.00000 0.00000 0.000000 0.00000 0.00000 0.00000
## 08 0.000000 0.000000 0.00000 0.00000 0.000000 0.00000 0.00000 0.00000
## 09 0.000000 0.000000 0.00000 0.00000 0.000000 0.00000 0.00000 0.00000
## 10 0.000000 0.000000 0.00000 0.00000 0.000000 0.00000 0.00000 0.00000
##     L1.09  L1.10   L1.11   L1.12  L1.13   L1.14    L1.15   L1.16
## 01 0.2419 0.1912 0.06221 0.09447 0.1014 0.02535 0.006912 0.01613
## 02 0.2419 0.1912 0.06221 0.09447 0.1014 0.02535 0.006912 0.01613
## 03 0.5000 0.0000 0.00000 0.00000 0.5000 0.00000 0.000000 0.00000
## 04 0.5000 0.0000 0.00000 0.00000 0.0000 0.00000 0.000000 0.00000
## 05 0.5000 0.0000 0.00000 0.00000 0.0000 0.00000 0.000000 0.00000
## 06 0.5000 0.0000 0.00000 0.00000 0.5000 0.00000 0.000000 0.00000
## 07 1.0000 0.0000 0.00000 0.00000 0.0000 0.00000 0.000000 0.00000
## 08 0.5000 0.0000 0.00000 0.00000 0.5000 0.00000 0.000000 0.00000
## 09 0.0000 0.5000 0.00000 0.00000 0.5000 0.00000 0.000000 0.00000
## 10 1.0000 0.0000 0.00000 0.00000 0.0000 0.00000 0.000000 0.00000</code></pre>
<p>Notice that there are a lot more alleles than there were originally. This is because the procedure is performed over the entire data set, not by population. Let’s look at what happens if we perform the same routine on the subsetted data.</p>
<pre class="r"><code>nan1 &lt;- popsub(nancycats, 1, drop = TRUE) # dropping alleles not in this population. 
nan1mean &lt;- missingno(nan1, &quot;mean&quot;)</code></pre>
<pre><code>## 
##  Replaced 8 missing values</code></pre>
<pre class="r"><code>nan1mean[loc = &quot;L1&quot;]@tab</code></pre>
<pre><code>##     L1.1   L1.2   L1.3 L1.4
## 01 0.125 0.5625 0.0625 0.25
## 02 0.125 0.5625 0.0625 0.25
## 03 0.000 0.5000 0.0000 0.50
## 04 0.500 0.5000 0.0000 0.00
## 05 0.500 0.5000 0.0000 0.00
## 06 0.000 0.5000 0.0000 0.50
## 07 0.000 1.0000 0.0000 0.00
## 08 0.000 0.5000 0.0000 0.50
## 09 0.000 0.0000 0.5000 0.50
## 10 0.000 1.0000 0.0000 0.00</code></pre>
</div>
<div id="hardy-weinberg-equilibrium" class="section level2">
<h2>Hardy-Weinberg equilibrium</h2>
<p>Next, let’s determine if our populations are in Hardy-Weinberg equilibrium. We will again use the nancycats data to test for HWE using genind objects, a dataframe particular to the <em>adegenet</em> R package:</p>
<pre class="r"><code>if (!require(genetics)) 
  install.packages(&quot;genetics&quot;, repos = &quot;http://cran.at.r-project.org&quot;)
nanhwe.full &lt;- HWE.test.genind(nancycats) # returns a list
nanhwe.mat  &lt;- HWE.test.genind(nancycats, res.type = &quot;matrix&quot;) # returns a matrix</code></pre>
<p>The full report (<code>nanhwe.full</code>) will give us a list of loci, each containing a list of populations, where each population is the result of a <span class="math">\(\chi^2\)</span> test. We will analyze data for a few loci to avoid writing a large amount of data to the screen. With the following statements, we can see that at locus1 (<code>fca8</code>), we have no good evidence to reject the null hypothesis that population 1 (<code>P01</code>) is not in HWE, but we have evidence that population 2 (<code>P02</code>) is not in HWE:</p>
<pre class="r"><code>nanhwe.full$fca8$P01</code></pre>
<pre><code>## 
##  Pearson&#39;s Chi-squared test
## 
## data:  tab
## X-squared = 5.136, df = 6, p-value = 0.5265</code></pre>
<pre class="r"><code>nanhwe.full$fca8$P02</code></pre>
<pre><code>## 
##  Pearson&#39;s Chi-squared test
## 
## data:  tab
## X-squared = 29.86, df = 15, p-value = 0.01243</code></pre>
<p>Of course, checking all of these is tedious, but we can look at a matrix for all loci by population:</p>
<pre class="r"><code>nanhwe.mat</code></pre>
<pre><code>##         fca8     fca23     fca43   fca45     fca77    fca78     fca90
## P01 0.526517 3.674e-03 0.2280402 0.43751 0.0208150 1.000000 1.765e-01
## P02 0.012427 9.955e-03 0.0001328 0.66252 0.0002153 0.073976 9.040e-07
## P03 0.008042 1.916e-02 0.5099205 0.90711 0.3460930 0.007422 4.882e-01
## P04 0.030857 3.234e-01 0.7685881 0.18515 0.0259008 0.134971 2.425e-04
## P05 0.395499 1.851e-01 0.2095530 0.65080 0.5544311 0.001087 4.948e-02
## P06 0.035773 3.720e-02 0.3933427 0.02992 0.0564054 0.166577 1.441e-01
## P07 0.001015 1.269e-02 0.9591441 0.45935 0.4482332 0.000760 2.556e-01
## P08 0.916112 3.966e-01 0.0919880 0.31833 0.0087894 0.032459 3.067e-01
## P09 0.315106 2.874e-02 0.6457715 0.16271 0.7313058 0.123889 1.948e-02
## P10 0.058729 8.743e-01 0.3820891 0.25979 0.0051378 0.152917 3.720e-02
## P11 0.155066 8.492e-02 0.1718178 0.46374 0.0535610 0.131432 6.895e-03
## P12 0.164120 2.434e-02 0.0014486 0.99994 0.2473303 0.394489 1.450e-01
## P13 0.263065 7.554e-04 0.0166925 0.69861 0.0782627 0.463145 6.731e-02
## P14 0.425618 1.298e-05 0.0613204 0.63565 0.0237132 0.020968 8.541e-02
## P15 0.705557 5.149e-01 0.5669695 0.78928 0.7006925 0.004625 2.563e-01
## P16 0.281227 4.745e-02 0.4923883 0.43651 0.2143387 0.002940 1.088e-01
## P17 0.105208 3.870e-01 0.0147464      NA 0.2844190 0.622681 3.508e-01
##         fca96     fca37
## P01 3.808e-01 1.728e-02
## P02 4.067e-13 5.889e-01
## P03 5.619e-01 3.600e-02
## P04 8.593e-01 5.750e-01
## P05 4.642e-01 3.896e-05
## P06 3.200e-01 7.168e-01
## P07 6.791e-01 1.943e-01
## P08 6.613e-01 4.085e-02
## P09 8.324e-01 7.067e-01
## P10 8.985e-01 6.182e-01
## P11 7.234e-02 7.412e-02
## P12 9.953e-01 5.611e-02
## P13 4.214e-01 1.385e-01
## P14 1.129e-02 9.212e-01
## P15 3.299e-02 5.719e-02
## P16 6.018e-02 5.609e-01
## P17 1.000e+00 1.648e-01</code></pre>
<p>However, this is still hard to sift through. We will thus define an <span class="math">\(\alpha\)</span> value and set everything above that value to 1 so that we can visually detect candidate loci where we might reject the <span class="math">\(H_o\)</span> of HWE.</p>
<pre class="r"><code>alpha  &lt;- 0.05
newmat &lt;- nanhwe.mat
newmat[newmat &gt; alpha] &lt;- 1</code></pre>
<p>Now we can create a simple heatmap with <code>levelplot</code>.</p>
<pre class="r"><code>library(lattice)
levelplot(newmat)</code></pre>
<p><img src="C5_Locus_Stats_files/figure-html/unnamed-chunk-13.png" title="Heatmap showing significant departures from HWE" alt="Heatmap showing significant departures from HWE" width="700px" /></p>
<p>This simple plot shows us loci in rows and populations in columns. Note, that all loci shown in pink are loci suspected for not being in HWE with <span class="math">\(p \leq 0.05\)</span>.</p>
</div>
<div id="references" class="section level2">
<h2>References</h2>
<p> 
EM Goss, JF Tabima, DEL Cooke, S Restrepo, WE Fry, GA Forbes, VJ Fieland, M Cardenas, NJ Grünwald, (2014) The Irish famine pathogen Phytophthora infestans originated in central Mexico rather than the Andes. <em>Proceedings of the National Academy of Sciences</em> <strong>111</strong> (24) 8791-8796-NA <a href="http://grunwaldlab.cgrb.oregonstate.edu/irish-potato-famine-pathogen-phytophthora-infestans-originated-central-mexico-rather-andes">http://grunwaldlab.cgrb.oregonstate.edu/irish-potato-famine-pathogen-phytophthora-infestans-originated-central-mexico-rather-andes</a>
</p>


</div>

</div>
</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
$(document).ready(function () {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
});

</script>

<!-- dynamically load mathjax for compatibility with --self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "libs/mathjax-2.3.0/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
