<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta http-equiv="Content-Style-Type" content="text/css" />
<meta name="generator" content="pandoc" />



<title>9: AMOVA</title>

<script src="libs/jquery-1.11.0/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1.0" />
<link href="libs/bootstrap-2.3.2/css/readable.min.css" rel="stylesheet" />
<link href="libs/bootstrap-2.3.2/css/bootstrap-responsive.min.css" rel="stylesheet" />
<script src="libs/bootstrap-2.3.2/js/bootstrap.min.js"></script>
<meta name="robots" content="noindex" />
<style type="text/css">

/* padding for bootstrap navbar */
body {
  padding-top: 50px;
  padding-bottom: 40px;
}
@media (max-width: 979px) {
  body {
    padding-top: 0;
  }
}

/* offset scroll position for anchor links (for fixed navbar)  */
@media (min-width: 980px) {
  .section h2 {
    padding-top: 52px;
    margin-top: -52px;
  }
  .section h3 {
    padding-top: 52px;
    margin-top: -52px;
  }
}


/* don't use link color in navbar */
.dropdown-menu>li>a {
  color: black;
}

</style>

<style type="text/css">code{white-space: pre;}</style>
<link rel="stylesheet"
      href="libs/highlight/default.css"
      type="text/css" />
<script src="libs/highlight/highlight.js"></script>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs && document.readyState && document.readyState === "complete") {
   window.setTimeout(function() {
      hljs.initHighlighting();
   }, 0);
}
</script>



</head>

<body>

<div class="container">


<div class="navbar navbar-default navbar-fixed-top">
  <div class="navbar-inner">
    <div class="container">
      <button type="button" class="btn btn-navbar" data-toggle="collapse" data-target=".nav-collapse">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="brand" href="index.html"><b>Population Genetics in R</b></a>
      <div class="nav-collapse collapse">
        <ul class="nav">
          <li><a href="TOC.html">Home</a></li>
          <li class="dropdown">
            <a href="chapters" class="dropdown-toggle" data-toggle="dropdown">Chapters<b class="caret"></b></a>
            <ul class="dropdown-menu">
               <li><a href="Foreword.html">i. Preface</a></li>
               <li><a href="C1_Introduction.html">1. Introduction</a></li>
               <li><a href="C2_Getting_ready_to_use_R.html">2. Getting ready to use R</a></li>
               <li><a href="C3_Data_format_loading.html">3. First steps</a></li>
               <li><a href="C4_Population_Hierarchies.html">4. Population hierarchies</a></li>
               <li><a href="C5_Locus_Stats.html">5. Locus-based statistics and missing data</a></li>
               <li><a href="C6_Genotypic_EvenRichDiv.html">6. Genotypic evenness, richness, and diversity</a></li>
               <li><a href="C7_Linkage_disequilibrium.html">7. Linkage disequilibrium</a></li>
               <li><a href="C8_Pop_Structure.html">8. Population structure</a></li>
               <li><a href="C9_AMOVA.html">9. AMOVA</a></li>
               <li><a href="C10_DAPC.html">10. Discriminant analysis of principal components (DAPC)</a></li>
               </ul>
            </li>
          <li class="dropdown">
            <a href="appendices" class="dropdown-toggle" data-toggle="dropdown">Appendices<b class="caret"></b></a>
            <ul class = "dropdown-menu">
               <li><a href="A1_Data_sets.html">A1. Data sets</a></li>
               <li><a href="A2_Function_Glossary.html">A2. Function glossary</a></li>
            </ul>
          </li>
          
        <ul class="nav">
          <li><a href="Authors.html">About</a></li>
          <li class="dropdown">

        </ul>
      </div><!--/.nav-collapse -->
    </div>
  </div>
</div>

<div class="container">
<div class="row">
<div class="span10 offset1">

<div id="header">
<h1 class="title">9: AMOVA</h1>
</div>

<div id="TOC">
<ul>
<li><a href="#introduction">Introduction</a></li>
<li><a href="#data-set">Data set</a></li>
<li><a href="#analysis">Analysis</a></li>
<li><a href="#significance-testing">Significance testing</a></li>
<li><a href="#random-population-structure">Random population structure</a></li>
<li><a href="#conclusions">Conclusions</a></li>
<li><a href="#references">References</a></li>
</ul>
</div>

<div id="introduction" class="section level2">
<h2>Introduction</h2>
<p>In this chapter, we will utilize AMOVA to analyze our populations. AMOVA stands for <strong>A</strong>nalysis of <strong>MO</strong>lecular <strong>VA</strong>riance and is a method to detect population differentiation utilizing molecular markers (<a href="http://www.genetics.org/content/131/2/479.abstract">Excoffier et al. 1992</a>). This procedure was initially implemented for DNA haplotypes, but applies to any marker system. <em>Poppr</em>’s implementation of AMOVA requires two very basic components: A distance matrix derived from the data and a separate table used to partition the data into different hierarchical levels.</p>
<p>The distance matrix can be calculated using any distance as long as it is euclidean. The distance that is used in the program Arlequin is the opposite of the Kronecker Delta function that counts the number of differences summed over <span class="math">\(L\)</span> loci:</p>
<p><span class="math">\[
\delta_{l,m} = \begin{cases}
1 \text{ if } m = l,\\
0 \text{ if } m \neq l
\end{cases}
\]</span> <span class="math">\[
d_{i,j} = \sum_{L = 1}^L 1 - \delta_{i,j}
\]</span></p>
</div>
<div id="data-set" class="section level2">
<h2>Data set</h2>
<p>To calculate AMOVA in <em>poppr</em>, one simply needs to supply a data set with hierarchical levels. We will use the <em>Aphanomyces euteiches</em> data set from (<a href="http://apsjournals.apsnet.org/doi/abs/10.1094/PHYTO-96-1134">Grünwald &amp; Hoheisel, 2006</a>).</p>
<pre class="r"><code>library(poppr)
data(Aeut)
Aeut &lt;- as.genclone(Aeut)
Aeut</code></pre>
<pre><code>## 
## This is a genclone object
## -------------------------
## Genotype information:
## 
##    119 multilocus genotypes
##    187 diploid individuals
##     56 dominant loci
## 
## Population information:
## 
##      3 hierarchical levels - Pop_Subpop Pop Subpop
##      2 populations defined - Athena Mt. Vernon</code></pre>
<p>We can see that this data set contains clonal data and has three hierarchical levels where the first is really a combination of the other levels. We can take a look at the different hierarchical levels, first populations followed by subpopulations:</p>
<pre class="r"><code>table(gethierarchy(Aeut, ~Pop))</code></pre>
<pre><code>## 
##     Athena Mt. Vernon 
##         97         90</code></pre>
<pre class="r"><code>table(gethierarchy(Aeut, ~Pop/Subpop, combine = FALSE))</code></pre>
<pre><code>##             Subpop
## Pop           1  2  3  4  5  6  7  8  9 10
##   Athena      9 12 10 13 10  5 11  8 10  9
##   Mt. Vernon 10  6  8 12 17 12 12 13  0  0</code></pre>
<p>In this example, we have a data set of 187 individuals sampled from two fields in regions Athena and Mt. Vernon over different 8 or 10 soil samples within each field. We want to see if most of the variance is observed at the sample, field, or regional level.</p>
</div>
<div id="analysis" class="section level2">
<h2>Analysis</h2>
<p>In panmictic populations, we would expect to see most of the variance arise from within samples. If we see that the most of the variance occurs among samples within populations or among populations, then there is evidence that we have some sort of population structure. In the case of clonal organisms, this would help support a hypothesis of clonal reproduction. Since <em>Aphanomyces eutieches</em> is known to be clonal, we would not expect most of the variance to come from within samples.</p>
<p>Let’s invoke the AMOVA functions with and without clone correction:</p>
<pre class="r"><code>Aeutamova &lt;- poppr.amova(Aeut, ~Pop/Subpop)
Aeutamovacc &lt;- poppr.amova(Aeut, ~Pop/Subpop, clonecorrect = TRUE)</code></pre>
<p>We’ll look at the AMOVA results for both analyses.</p>
<pre class="r"><code>Aeutamova</code></pre>
<pre><code>## $call
## ade4::amova(samples = xtab, distances = xdist, structures = xstruct)
## 
## $results
##                             Df Sum Sq  Mean Sq
## Between Pop                  1 1051.2 1051.235
## Between samples Within Pop  16  273.5   17.091
## Within samples             169  576.5    3.411
## Total                      186 1901.2   10.221
## 
## $componentsofcovariance
##                                         Sigma       %
## Variations  Between Pop                11.063  70.007
## Variations  Between samples Within Pop  1.329   8.407
## Variations  Within samples              3.411  21.586
## Total variations                       15.803 100.000
## 
## $statphi
##                      Phi
## Phi-samples-total 0.7841
## Phi-samples-Pop   0.2803
## Phi-Pop-total     0.7001</code></pre>
<pre class="r"><code>Aeutamovacc</code></pre>
<pre><code>## $call
## ade4::amova(samples = xtab, distances = xdist, structures = xstruct)
## 
## $results
##                             Df Sum Sq Mean Sq
## Between Pop                  1  742.0 741.987
## Between samples Within Pop  16  185.7  11.605
## Within samples             123  520.1   4.229
## Total                      140 1447.8  10.341
## 
## $componentsofcovariance
##                                          Sigma       %
## Variations  Between Pop                10.4132  66.778
## Variations  Between samples Within Pop  0.9521   6.105
## Variations  Within samples              4.2286  27.117
## Total variations                       15.5938 100.000
## 
## $statphi
##                      Phi
## Phi-samples-total 0.7288
## Phi-samples-Pop   0.1838
## Phi-Pop-total     0.6678</code></pre>
<p>The first thing to look at are the <code>$results</code> element. The degrees of freedom (the <code>Df</code> column) should match what we expect from our (not clone-corrected) data. The number in the <code>Total</code> row should equal 186 or <span class="math">\(N - 1\)</span>, where values are calculated from pooled data. Note that here, “samples” actually refers to subpopulations since we cannot asses within-sample variance of dominant data.</p>
<p>The <code>$componentsofcovariance</code> table shows how much variance is detected at each hierarchical level. We expect variations within samples to give the greatest amount of variation for populations that are not significantly differentiated. <code>Sigma</code> represents the variance, <span class="math">\(\sigma\)</span>, for each hierarchical level and to the right is the percent of the total.</p>
<p>Finally, <code>$statphi</code> provides the <span class="math">\(\phi\)</span> population differentiation statistics. These are used to test hypotheses about population differentiation. We would expect a higher <span class="math">\(\phi\)</span> statistic to represent a higher amount of differentiation.</p>
<p>Note, if you want to make a table of any of these components, you can isolate them by using the <code>$</code> operator and then export it to a table with <code>write.table</code>. Here’s an example with the components of covariance:</p>
<pre class="r"><code>write.table(Aeutamova$componentsofcovariance, sep = &quot;,&quot;, file = &quot;~/Documents/AeutiechesAMOVA.csv&quot;)</code></pre>
</div>
<div id="significance-testing" class="section level2">
<h2>Significance testing</h2>
<p>To test if populations are significantly different, we perform a randomization test using the function <code>randtest()</code> from the <em>ade4</em> package. This will randomly permute the sample matrices as described in (<a href="http://www.genetics.org/content/131/2/479.abstract">Excoffier et al. 1992</a>).</p>
<pre class="r"><code>set.seed(1999)
Aeutsignif   &lt;- randtest(Aeutamova, nrepet = 999)
Aeutccsignif &lt;- randtest(Aeutamovacc, nrepet = 999)</code></pre>
<pre class="r"><code>plot(Aeutsignif)</code></pre>
<p><img src="C9_AMOVA_files/figure-html/Aeut_random_plot.png" title="plot of chunk Aeut_random_plot" alt="plot of chunk Aeut_random_plot" width="700px" /></p>
<pre class="r"><code>Aeutsignif</code></pre>
<pre><code>## class: krandtest 
## Monte-Carlo tests
## Call: randtest.amova(xtest = Aeutamova, nrepet = 999)
## 
## Number of tests:   3 
## 
## Adjustment method for multiple comparisons:   none 
## Permutation number:   999 
##                         Test    Obs Std.Obs   Alter Pvalue
## 1  Variations within samples  3.411 -31.385    less  0.001
## 2 Variations between samples  1.329  20.687 greater  0.001
## 3     Variations between Pop 11.063   9.425 greater  0.001
## 
## other elements: adj.method call</code></pre>
<p>From this output, you can see three histograms representing the distribution of the randomized hierarchy. The black line represents the observed data. You can see a table of observed results in the output showing that there is significant population structure considering all levels of the population hierarchy. Of course, this could be due to the presence of clones, so let’s visualize the results from the clone corrected data set below:</p>
<pre class="r"><code>plot(Aeutccsignif)</code></pre>
<p><img src="C9_AMOVA_files/figure-html/Aeut_clonecorrect_random_plot.png" title="plot of chunk Aeut_clonecorrect_random_plot" alt="plot of chunk Aeut_clonecorrect_random_plot" width="700px" /></p>
<pre class="r"><code>Aeutccsignif</code></pre>
<pre><code>## class: krandtest 
## Monte-Carlo tests
## Call: randtest.amova(xtest = Aeutamovacc, nrepet = 999)
## 
## Number of tests:   3 
## 
## Adjustment method for multiple comparisons:   none 
## Permutation number:   999 
##                         Test     Obs Std.Obs   Alter Pvalue
## 1  Variations within samples  4.2286 -20.901    less  0.001
## 2 Variations between samples  0.9521   9.560 greater  0.001
## 3     Variations between Pop 10.4132   9.758 greater  0.001
## 
## other elements: adj.method call</code></pre>
<p>The above graphs show significant population differentiation at all levels given that the observed <span class="math">\(\phi\)</span> does not fall within the distribution expected from the permutation. Compare the results of our AMOVA analysis to those published in (<a href="http://apsjournals.apsnet.org/doi/abs/10.1094/PHYTO-96-1134">Grünwald &amp; Hoheisel, 2006</a>). They should be identical.</p>
</div>
<div id="random-population-structure" class="section level2">
<h2>Random population structure</h2>
<p>Since AMOVA is used to detect whether or not there is significant population structure, we can see what happens when we randomly shuffle the population assignments in our data. Here we will show what the populations look like before and after shuffling:</p>
<pre class="r"><code>Aeut.new &lt;- Aeut
head(gethierarchy(Aeut)[, -1])</code></pre>
<pre><code>##      Pop Subpop
## 1 Athena      1
## 2 Athena      1
## 3 Athena      1
## 4 Athena      1
## 5 Athena      1
## 6 Athena      1</code></pre>
<pre class="r"><code>set.seed(9001)
head(gethierarchy(Aeut)[sample(nInd(Aeut)), -1])</code></pre>
<pre><code>##            Pop Subpop
## 44      Athena      4
## 177 Mt. Vernon      8
## 36      Athena      4
## 127 Mt. Vernon      4
## 8       Athena      1
## 39      Athena      4</code></pre>
<p>Here we see that the populations are completely shuffled, so in the next step, we will reassign the hierarchy with these newly shuffled populations and rerun the AMOVA analysis.</p>
<pre class="r"><code>set.seed(9001)
sethierarchy(Aeut.new) &lt;- gethierarchy(Aeut)[sample(nInd(Aeut)), -1]
Aeut.new.amova         &lt;- poppr.amova(Aeut.new, ~Pop/Subpop)</code></pre>
<pre><code>## 
##  No missing values detected.</code></pre>
<pre class="r"><code>Aeut.new.amova</code></pre>
<pre><code>## $call
## ade4::amova(samples = xtab, distances = xdist, structures = xstruct)
## 
## $results
##                             Df  Sum Sq Mean Sq
## Between Pop                  1   23.21  23.206
## Between samples Within Pop  16  136.34   8.521
## Within samples             169 1741.65  10.306
## Total                      186 1901.20  10.221
## 
## $componentsofcovariance
##                                          Sigma       %
## Variations  Between Pop                 0.1589   1.544
## Variations  Between samples Within Pop -0.1733  -1.684
## Variations  Within samples             10.3056 100.140
## Total variations                       10.2912 100.000
## 
## $statphi
##                         Phi
## Phi-samples-total -0.001402
## Phi-samples-Pop   -0.017106
## Phi-Pop-total      0.015440</code></pre>
<pre class="r"><code>Aeut.new.amova.test    &lt;- randtest(Aeut.new.amova, nrepet = 999)
Aeut.new.amova.test</code></pre>
<pre><code>## class: krandtest 
## Monte-Carlo tests
## Call: randtest.amova(xtest = Aeut.new.amova, nrepet = 999)
## 
## Number of tests:   3 
## 
## Adjustment method for multiple comparisons:   none 
## Permutation number:   999 
##                         Test     Obs Std.Obs   Alter Pvalue
## 1  Variations within samples 10.3056  0.3706    less  0.602
## 2 Variations between samples -0.1733 -0.7517 greater  0.775
## 3     Variations between Pop  0.1589  1.4471 greater  0.096
## 
## other elements: adj.method call</code></pre>
<pre class="r"><code>plot(Aeut.new.amova.test)</code></pre>
<p><img src="C9_AMOVA_files/figure-html/randomized_hierarchy_plot.png" title="plot of chunk randomized_hierarchy_plot" alt="plot of chunk randomized_hierarchy_plot" width="700px" /></p>
<p>We see that there now is no significant population structure.</p>
</div>
<div id="conclusions" class="section level2">
<h2>Conclusions</h2>
<p>AMOVA is a powerful tool that can help support hypotheses of population structure due to clonal reproduction or isolation without making assumptions about Hardy-Weinberg equilibrium. We have shown that we can reject the <span class="math">\(H_o\)</span> of random mating between the two populations and have strong evidence that these populations are significantly differentiated at all hierarchical levels (<a href="http://apsjournals.apsnet.org/doi/abs/10.1094/PHYTO-96-1134">Grünwald &amp; Hoheisel, 2006</a>). From these results, we can investigate hypotheses as to why these populations are significantly differentiated.</p>
<p>This example was performed with a data set of dominant (AFLP) markers, but it can also be performed on codominant markers as well (SNP, SSR). These provide more information because within sample (individual) variance is also assessed. If one wants to utilize a genetic distance that has biological relevance, a different distance matrix can be provided by the user. See <code>help('amova', package = 'poppr')</code> for more details.</p>
</div>
<div id="references" class="section level2">
<h2>References</h2>
<p> 
L Excoffier, PE Smouse, JM Quattro, (1992) Analysis of molecular variance inferred from metric distances among DNA haplotypes: application to human mitochondrial DNA restriction data. <em>Genetics</em> <strong>131</strong> (2) 479-491 <a href="http://www.genetics.org/content/131/2/479.abstract">http://www.genetics.org/content/131/2/479.abstract</a>
</p><p> 
NJ Grünwald, G-A Hoheisel, (2006) Hierarchical analysis of diversity, selfing, and genetic differentiation in populations of the oomycete Aphanomyces euteiches. <em>Phytopathology</em> <strong>96</strong> (10) 1134-1141 <a href="http://apsjournals.apsnet.org/doi/abs/10.1094/PHYTO-96-1134">http://apsjournals.apsnet.org/doi/abs/10.1094/PHYTO-96-1134</a>
</p>



</div>

</div>
</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
$(document).ready(function () {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
});

</script>

<!-- dynamically load mathjax for compatibility with --self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "libs/mathjax-2.3.0/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
