<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta http-equiv="Content-Style-Type" content="text/css" />
<meta name="generator" content="pandoc" />



<title>9: Population structure: G_{ST}, clustering and minimum spanning networks</title>

<script src="libs/jquery-1.11.0/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1.0" />
<link href="libs/bootstrap-2.3.2/css/readable.min.css" rel="stylesheet" />
<link href="libs/bootstrap-2.3.2/css/bootstrap-responsive.min.css" rel="stylesheet" />
<script src="libs/bootstrap-2.3.2/js/bootstrap.min.js"></script>
<style type="text/css">

/* padding for bootstrap navbar */
body {
  padding-top: 50px;
  padding-bottom: 40px;
}
@media (max-width: 979px) {
  body {
    padding-top: 0;
  }
}

/* offset scroll position for anchor links (for fixed navbar)  */
@media (min-width: 980px) {
  .section h2 {
    padding-top: 52px;
    margin-top: -52px;
  }
  .section h3 {
    padding-top: 52px;
    margin-top: -52px;
  }
}


/* don't use link color in navbar */
.dropdown-menu>li>a {
  color: black;
}

</style>

<style type="text/css">code{white-space: pre;}</style>
<link rel="stylesheet"
      href="libs/highlight/default.css"
      type="text/css" />
<script src="libs/highlight/highlight.js"></script>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs && document.readyState && document.readyState === "complete") {
   window.setTimeout(function() {
      hljs.initHighlighting();
   }, 0);
}
</script>



</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
</style>
<div class="container-fluid main-container">


<div class="navbar navbar-default navbar-fixed-top">
  <div class="navbar-inner">
    <div class="container">
      <button type="button" class="btn btn-navbar" data-toggle="collapse" data-target=".nav-collapse">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="brand" href="index.html"><b>Population Genetics in R</b></a>
      <div class="nav-collapse collapse">
        <ul class="nav">
          <!--><li><a href="TOC.html">Home</a></li><-->
          <li class="dropdown">
            <a href="chapters" class="dropdown-toggle" data-toggle="dropdown">Chapters<b class="caret"></b></a>
            <ul class="dropdown-menu">
               <li><a href="Preface.html">i. Preface</a></li>
               <li><a href="C1_Introduction.html">1. Introduction</a></li>
               <li><a href="C2_Getting_ready_to_use_R.html">2. Getting ready to use R</a></li>
               <li><a href="C3_Data_Preparation.html">3. Data Preparation</a></li>
               <li><a href="C4_First_Steps.html">4. First steps</a></li>
               <li><a href="C5_Population_Hierarchies.html">5. Population hierarchies</a></li>
               <li><a href="C6_Locus_Stats.html">6. Locus-based statistics and missing data</a></li>
               <li><a href="C7_Genotypic_EvenRichDiv.html">7. Genotypic evenness, richness, and diversity</a></li>
               <li><a href="C8_Linkage_disequilibrium.html">8. Linkage disequilibrium</a></li>
               <li><a href="C9_Pop_Structure.html">9. Population structure</a></li>
               <li><a href="C10_AMOVA.html">10. AMOVA</a></li>
               <li><a href="C11_DAPC.html">11. Discriminant analysis of principal components (DAPC)</a></li>
               </ul>
            </li>
          <li class="dropdown">
            <a href="appendices" class="dropdown-toggle" data-toggle="dropdown">Appendices<b class="caret"></b></a>
            <ul class = "dropdown-menu">
               <li><a href="A1_Data_sets.html">A1. Data sets</a></li>
               <li><a href="A2_Function_Glossary.html">A2. Function glossary</a></li>
            </ul>
          </li>
          
        <ul class="nav">
          <li><a href="Authors.html">About</a></li>
          <li class="dropdown">

        </ul>
      </div><!--/.nav-collapse -->
    </div>
  </div>
</div>

<div class="container">
<div class="row">
<div class="span10 offset1">

<div id="header">
<h1 class="title">9: Population structure: <span class="math">\(G_{ST}\)</span>, clustering and minimum spanning networks</h1>
</div>

<div id="TOC">
<ul>
<li><a href="#g_st-an-example-with-felis-catus-data."><span class="math">\(G_{ST}\)</span> an example with <em>Felis catus</em> data.</a></li>
<li><a href="#minimum-spanning-network">Minimum spanning network</a></li>
<li><a href="#k-means-hierarchical-clustering">K-means hierarchical clustering</a></li>
<li><a href="#other-methods">Other methods</a></li>
<li><a href="#references">References</a></li>
</ul>
</div>

<p>In this chapter we explore various ways of assessing if population are structured (e.g., differentiated). You can think of population structure as identifying clusters or groups of more closely related individuals resulting from reduced gene flow among these groups. Populations can be studied to determine if they are structured by using for example population differentiation <span class="math">\(G_{ST}\)</span>, clustering, or graphing of minimum spanning networks (MSN). Let’s first look at an example of population differentiation.</p>
<div id="g_st-an-example-with-felis-catus-data." class="section level2">
<h2><span class="math">\(G_{ST}\)</span> an example with <em>Felis catus</em> data.</h2>
<p>Assessing genetic diversity almost always starts with an analysis of a parameter such as <span class="math">\(G_{ST}\)</span>. There are <a href="http://www.molecularecologist.com/2011/03/should-i-use-fst-gst-or-d-2/">lengthy debates</a> as to what measure of differentiation is better (<a href="http://onlinelibrary.wiley.com/doi/10.1111/j.1755-0998.2010.02927.x/full">Meirmans &amp; Hedrick, 2011</a>). Instead of going into that lengthy debate, it would be more worthwhile to point you into the direction of a package dedicated to Modern Methods of Differentiation called <em>mmod</em>. We will use the data set <em>nancycats</em> containing 17 colonies of cats collected from Nancy, France. As cats tend to stay within small groups, we expect to see some population differentiation. In terms of these diversity measures, an index of 0 indicates no differentiation, whereas 1 indicates that populations are segregating for differing alleles.</p>
<p>Let’s load the package and the example data set.</p>
<pre class="r"><code>if (!require(&quot;mmod&quot;)){
  install.packages(&quot;mmod&quot;, repos = &quot;http://cran.at.r-project.org&quot;)
  require(&quot;mmod&quot;)
}
data(nancycats)
nancycats</code></pre>
<pre><code>## 
##    #####################
##    ### Genind object ### 
##    #####################
## - genotypes of individuals - 
## 
## S4 class:  genind
## @call: genind(tab = truenames(nancycats)$tab, pop = truenames(nancycats)$pop)
## 
## @tab:  237 x 108 matrix of genotypes
## 
## @ind.names: vector of  237 individual names
## @loc.names: vector of  9 locus names
## @loc.nall: number of alleles per locus
## @loc.fac: locus factor for the  108 columns of @tab
## @all.names: list of  9 components yielding allele names for each locus
## @ploidy:  2
## @type:  codom
## 
## Optional contents: 
## @pop:  factor giving the population of each individual
## @pop.names:  factor giving the population of each individual
## 
## @other: a list containing: xy</code></pre>
<p>Now we will use Hendrick’s standardized <span class="math">\(G_{ST}\)</span> to assess population structure among these populations (<a href="http://onlinelibrary.wiley.com/doi/10.1111/j.0014-3820.2005.tb01814.x/abstract">Hedrick, 2005</a>).</p>
<pre class="r"><code>Gst_Hedrick(nancycats)</code></pre>
<pre><code>## $per.locus
##     fca8    fca23    fca43    fca45    fca77    fca78    fca90    fca96 
##  0.47504  0.29567  0.26758 -0.05706  0.48558  0.19333  0.38076  0.39139 
##    fca37 
##  0.16096 
## 
## $global
## [1] 0.2821</code></pre>
<p>What does this output tell us?</p>
</div>
<div id="minimum-spanning-network" class="section level2">
<h2>Minimum spanning network</h2>
<p>For this section, we will use the <code>monpop</code> data set from (<a href="">Everhart &amp; Scherm, 2014</a>). see <a href="C6_Genotypic_EvenRichDiv.html">Chapter 6</a> for more details. We will be focusing on sources of multilocus genotypes. The life cycle of the pathogen <em>Monilinia fructicola</em> begins with an ascospore (sexual propagule) released from a mummified peach fruit that had overwintered on the ground. It infects an emerging blossom that, in turn, asexually infects fruit, which proceed with cyclical infections. We might have questions such as:</p>
<ol style="list-style-type: decimal">
<li>Are the major genotypes of Fruit Rot (FR) samples closely related?</li>
<li>To what degree do the Blossom Blight (BB) samples contribute to the FR?</li>
</ol>
<p>Let’s load the data:</p>
<pre class="r"><code>library(poppr)
data(monpop)
splithierarchy(monpop) &lt;- ~Tree/Year/Symptom
summary(monpop)</code></pre>
<pre><code>## 
##  # Total number of genotypes:  694 
## 
##  # Population sample sizes:  
##  7_09_BB 26_09_BB 26_09_FR  7_09_FR 26_10_BB 45_10_BB 79_10_BB 79_10_FR 
##       23       41      132       73        5       13        1       64 
## 26_10_FR 45_10_FR 26_11_BB 26_11_FR 
##       85      130       30       97 
## 
##  # Number of alleles per locus:  
## L01 L02 L03 L04 L05 L06 L07 L08 L09 L10 L11 L12 L13 
##   3   6   3  11   9   5   8   7   5   9   8  11  10 
## 
##  # Number of alleles per population:  
## 01 02 03 04 05 06 07 08 09 10 11 12 
## 48 53 58 48 37 40 13 44 60 64 60 63 
## 
##  # Percentage of missing data:  
## [1] 0.5099
## 
##  # Observed heterozygosity:  
## [1] 0
## 
##  # Expected heterozygosity:  
## [1] 0</code></pre>
<p>We notice that tree number 26 is the only one to have been sampled for all three years. Let’s use it as an example.</p>
<pre class="r"><code>setpop(monpop) &lt;- ~Tree
t26 &lt;- popsub(monpop, &quot;26&quot;)
setpop(t26) &lt;- ~Year/Symptom
t26</code></pre>
<pre><code>## 
## This is a genclone object
## -------------------------
## Genotype information:
## 
##    155 multilocus genotypes
##    390 haploid individuals
##     13 codominant loci
## 
## Population information:
## 
##      3 hierarchical levels - Tree Year Symptom
##      6 populations defined - 9_BB 9_FR 10_BB 10_FR 11_BB 11_FR</code></pre>
<p>Now that we have our tree, let’s calculate a MSN using Bruvo’s distance (<a href="http://dx.doi.org/10.1111/j.1365-294X.2004.02209.x">Bruvo et al. 2004</a>):</p>
<pre class="r"><code># Set up our repeat lengths and populations to analyze
reps &lt;- c(6, 2, 4, 4, 4, 4, 4, 3, 4, 4, 2, 2, 2)
sub9 &lt;- c(&quot;9_BB&quot;, &quot;9_FR&quot;)

# Run the tree
t26.9msn &lt;- bruvo.msn(t26, replen = reps, sublist = sub9, showplot = FALSE)

# Visualize
set.seed(9001)
plot_poppr_msn(t26, t26.9msn, inds = &quot;none&quot;, palette = cm.colors)</code></pre>
<p><img src="C9_Pop_Structure_files/figure-html/unnamed-chunk-4.png" title="plot of chunk unnamed-chunk-4" alt="plot of chunk unnamed-chunk-4" width="700px" /></p>
<p>We can see that the Blossom Blight in the tree heavily contributed to the major groups of MLGs found in the Fruit Rot.</p>
<blockquote>
<p><strong>Try it!</strong> See if you can produce similar graphs with the 2010 and 2011 populations.</p>
</blockquote>
</div>
<div id="k-means-hierarchical-clustering" class="section level2">
<h2>K-means hierarchical clustering</h2>
<p>A recent study reported that the origin of the potato late blight pathogen <em>Phytophthora infestans</em> lies in Mexico as opposed to South America (<a href="http://dx.doi.org/10.1073/pnas.1401884111">Goss et al. 2014</a>). We saw in the previous chapter that South American populations showed signatures of clonal reproduction while Mexican populations showed no evidence rejecting the null hypothesis of random mating. In this section, we will use K-means clustering in combination with bootstrapped dendrograms to see how well this pattern holds up. Clonal populations should have short terminal branch lengths and should cluster according to those branches. Panmictic populations will show no clear pattern. Let’s look at the data:</p>
<pre class="r"><code>data(Pinf)
Pinf</code></pre>
<pre><code>## 
## This is a genclone object
## -------------------------
## Genotype information:
## 
##    72 multilocus genotypes
##    86 tetraploid individuals
##    11 codominant loci
## 
## Population information:
## 
##     2 hierarchical levels - Continent Country
##     2 populations defined - South America North America</code></pre>
<p>First, we will perform a cluster analysis:</p>
<pre class="r"><code>MX &lt;- popsub(Pinf, &quot;North America&quot;)
MXclust &lt;- find.clusters(MX)</code></pre>
<p><img src="C9_Pop_Structure_files/figure-html/MXPCA.png" alt="MX_PCA" /></p>
<pre><code>## Choose the number PCs to retain (&gt;=1):</code></pre>
<pre><code>&gt; 50</code></pre>
<p>PC stands for principal component, are unitless transofrmations of your data that explaining the variance observed. For the purposes of <code>find.clusters</code>, we can keep as many as we want.</p>
<p><img src="C9_Pop_Structure_files/figure-html/MXCLUST.png" alt="MX_CLUSTER" /></p>
<pre><code>## Choose the number PCs to retain (&gt;=2:</code></pre>
<pre><code>&gt; 3</code></pre>
<p>BIC stands for “Bayesian Information Criterion”. The lower the BIC value, the better. On the x axis are the number of clusters. We see that there is a bend at 3 clusters, indicating that the data clusters optimally into three groups.</p>
<p>And now we can see the cluster assignments:</p>
<pre class="r"><code>MXclust</code></pre>
<pre><code>## $Kstat
## NULL
## 
## $stat
## NULL
## 
## $grp
##  PiMX01  PiMX02  PiMX03  PiMX04  PiMX05  PiMX06  PiMX07  PiMX10  PiMX11 
##       1       1       2       3       3       3       3       2       2 
##  PiMX12  PiMX13  PiMX14  PiMX15  PiMX16  PiMX17  PiMX18  PiMX19  PiMX20 
##       3       2       2       1       2       1       3       3       3 
##  PiMX21  PiMX22  PiMX23  PiMX24  PiMX25  PiMX26  PiMX27  PiMX28  PiMX29 
##       1       2       1       2       3       1       2       2       3 
##  PiMX30  PiMX40  PiMX41  PiMX42  PiMX43  PiMX44  PiMX45  PiMX46  PiMX47 
##       2       2       1       1       1       1       1       1       1 
##  PiMX48  PiMX49  PiMX50 PiMXT01 PiMXT02 PiMXT03 PiMXT04 PiMXT05 PiMXT06 
##       2       2       2       1       1       3       3       1       3 
## PiMXT07 PiMXt48 PiMXt68 
##       3       3       3 
## Levels: 1 2 3
## 
## $size
## [1] 17 15 16</code></pre>
<p>We will go through the same procedure for the South American population.</p>
<pre class="r"><code>SA &lt;- popsub(Pinf, &quot;South America&quot;)
SAclust &lt;- find.clusters(SA)</code></pre>
<p><img src="C9_Pop_Structure_files/figure-html/SAPCA.png" alt="SA_PCA" /></p>
<pre><code>## Choose the number PCs to retain (&gt;=1):</code></pre>
<pre><code>&gt; 30</code></pre>
<p><img src="C9_Pop_Structure_files/figure-html/SACLUST.png" alt="SA_CLUSTER" /></p>
<pre><code>## Choose the number PCs to retain (&gt;=2):</code></pre>
<pre><code>&gt; 4</code></pre>
<p>Notice here that there is no uptick in the curve. This indicates that there <a href="http://lists.r-forge.r-project.org/pipermail/adegenet-forum/2011-June/000303.html">might not be enough information in the data set to properly cluster</a>. We will go ahead by choosing the highest number of clusters:</p>
<div id="trees" class="section level3">
<h3>Trees</h3>
<p>Now we will build trees. We are using Bruvo’s distance since polyploids bias calculation of other distances:</p>
<pre class="r"><code>pinfreps &lt;- c(2, 2, 6, 2, 2, 2, 2, 2, 3, 3, 2)
MXtree &lt;- bruvo.boot(MX, replen = pinfreps, cutoff = 50, quiet = TRUE)</code></pre>
<p><img src="C9_Pop_Structure_files/figure-html/trees1.png" title="plot of chunk trees" alt="plot of chunk trees" width="700px" /></p>
<pre class="r"><code>SAtree &lt;- bruvo.boot(SA, replen = pinfreps, cutoff = 50, quiet = TRUE)</code></pre>
<p><img src="C9_Pop_Structure_files/figure-html/trees2.png" title="plot of chunk trees" alt="plot of chunk trees" width="700px" /></p>
<p>We see very long terminal branches in the MX tree. Let’s see how the groups we found with the clustering algorithm match up:</p>
<pre class="r"><code>library(ape)
cols &lt;- rainbow(4)
plot.phylo(MXtree, cex = 0.8, font = 2, adj = 0, tip.color = cols[MXclust$grp])
nodelabels(MXtree$node.label, adj = c(1.3, -0.5), frame = &quot;n&quot;, cex = 0.8, font = 3)
axisPhylo(3)</code></pre>
<p><img src="C9_Pop_Structure_files/figure-html/grouptree.png" title="plot of chunk grouptree" alt="plot of chunk grouptree" width="700px" /></p>
<p>You can see that the assigned clusters don’t necessarily group with the dendrogram clusters. Let’s see what happens when we view this with the South American population:</p>
<pre class="r"><code>plot.phylo(SAtree, cex = 0.8, font = 2, adj = 0, tip.color = cols[SAclust$grp])
nodelabels(SAtree$node.label, adj = c(1.3, -0.5), frame = &quot;n&quot;, cex = 0.8, font = 3)
axisPhylo(3)</code></pre>
<p><img src="C9_Pop_Structure_files/figure-html/grouptreeSA.png" title="plot of chunk grouptreeSA" alt="plot of chunk grouptreeSA" width="700px" /></p>
<p>Everything clusters together nicely, further supporting a non-panmictic population.</p>
</div>
</div>
<div id="other-methods" class="section level2">
<h2>Other methods</h2>
<p>Not all methods of clustering need to be based on Bruvos’s distance. There are others available. See <code>?nei.dist</code> for more details. The function <code>aboot</code> will perform bootstrap analysis on any distance given for individuals or populations, and <code>poppr.msn</code> will create minimum spanning networks given a genind object and a distance.</p>
</div>
<div id="references" class="section level2">
<h2>References</h2>
<p> 
R Bruvo, NK Michiels, TG D’Souza, H Schulenburg, (2004) A simple method for the calculation of microsatellite genotype distances irrespective of ploidy level. <em>Molecular Ecology</em> <strong>13</strong> (7) 2101-2106 <a href="http://dx.doi.org/10.1111/j.1365-294X.2004.02209.x">10.1111/j.1365-294X.2004.02209.x</a>
</p><p> 
SE Everhart, H Scherm, (2014) Fine-scale genetic structure of <i>Monilinia fructicola</i> during brown rot epidemics within individual peach tree canopies. <em>Phytopathology</em> <strong>##</strong> (#) submitted-NA
</p><p> 
EM Goss, JF Tabima, DEL Cooke, S Restrepo, WE Fry, GA Forbes, VJ Fieland, M Cardenas, NJ Grünwald, (2014) The Irish potato famine pathogen <i>Phytophthora infestans</i> originated in central Mexico rather than the Andes. <em>Proceedings of the National Academy of Sciences</em> <a href="http://dx.doi.org/10.1073/pnas.1401884111">10.1073/pnas.1401884111</a>
</p><p> 
PW Hedrick, (2005) A standardized genetic differentiation measure. <em>Evolution</em> <strong>59</strong> (8) 1633-1638 <a href="http://onlinelibrary.wiley.com/doi/10.1111/j.0014-3820.2005.tb01814.x/abstract">http://onlinelibrary.wiley.com/doi/10.1111/j.0014-3820.2005.tb01814.x/abstract</a>
</p><p> 
PG Meirmans, PW Hedrick, (2011) Assessing population structure: <span class="math">\(F_{ST}\)</span> and related measures. <em>Molecular Ecology Resources</em> <strong>11</strong> (1) 5-18 <a href="http://onlinelibrary.wiley.com/doi/10.1111/j.1755-0998.2010.02927.x/full">http://onlinelibrary.wiley.com/doi/10.1111/j.1755-0998.2010.02927.x/full</a>
</p>


</div>

</div>
</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
$(document).ready(function () {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
});

</script>

<!-- dynamically load mathjax for compatibility with --self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "libs/mathjax-2.3.0/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
