<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />



<title>A2. Function glossary</title>

<script src="libs/jquery-1.11.0/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1.0" />
<link href="libs/bootstrap-2.3.2/css/readable.min.css" rel="stylesheet" />
<link href="libs/bootstrap-2.3.2/css/bootstrap-responsive.min.css" rel="stylesheet" />
<script src="libs/bootstrap-2.3.2/js/bootstrap.min.js"></script>
<script>
  (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
  (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
  m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
  })(window,document,'script','//www.google-analytics.com/analytics.js','ga');

  ga('create', 'UA-52691601-2', 'auto');
  ga('send', 'pageview');

</script>

<style type="text/css">

/* padding for bootstrap navbar */
body {
  padding-top: 50px;
  padding-bottom: 40px;
  padding-right: 50px;
  padding-left: -50px;
}
@media (max-width: 979px) {
  body {
    padding-top: 0;
  }
}

/* offset scroll position for anchor links (for fixed navbar)  */
@media (min-width: 980px) {
  .section h2 {
    padding-top: 52px;
    margin-top: -52px;
  }
  .section h3 {
    padding-top: 52px;
    margin-top: -52px;
  }
}


/* don't use link color in navbar */
.dropdown-menu>li>a {
  color: black;
}

</style>

<style type="text/css">code{white-space: pre;}</style>
<link rel="stylesheet"
      href="libs/highlight/default.css"
      type="text/css" />
<script src="libs/highlight/highlight.js"></script>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs && document.readyState && document.readyState === "complete") {
   window.setTimeout(function() {
      hljs.initHighlighting();
   }, 0);
}
</script>



</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
</style>
<div class="container-fluid main-container">


<div class="navbar navbar-default navbar-fixed-top">
  <div class="navbar-inner">
    <div class="container">
      <button type="button" class="btn btn-navbar" data-toggle="collapse" data-target=".nav-collapse">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="brand" href="index.html"><b>Population Genetics in R</b></a>
      <div class="nav-collapse collapse">
        <ul class="nav">
          <li class="dropdown">
            <a href="chapters" class="dropdown-toggle" data-toggle="dropdown">Chapters<b class="caret"></b></a>
            <ol class="dropdown-menu">
               <li type = "i"><a href="Preface.html">Preface</a></li>
               <li type = "1" value = "1"><a href="Introduction.html">Introduction</a></li>
               <li><a href="Getting_ready_to_use_R.html">Getting ready to use R</a></li>
               <li><a href="Data_Preparation.html">Data preparation</a></li>
               <li><a href="First_Steps.html">First steps</a></li>
               <li><a href="Population_Hierarchies.html">Population hierarchies and clone correction</a></li>
               <li><a href="Locus_Stats.html">Locus-based statistics and missing data</a></li>
               <li><a href="Genotypic_EvenRichDiv.html">Genotypic evenness, richness, and diversity</a></li>
               <li><a href="Linkage_disequilibrium.html">Linkage disequilibrium</a></li>
               <li><a href="Pop_Structure.html">Population structure</a></li>
               <li><a href="AMOVA.html">AMOVA</a></li>
               <li><a href="DAPC.html">Discriminant analysis of principal components (DAPC)</a></li>
            </ol>
          </li>
          <li class="dropdown">
            <a href="appendices" class="dropdown-toggle" data-toggle="dropdown">Appendices<b class="caret"></b></a>
            <ul class = "dropdown-menu">
               <li><a href="Data_sets.html">A1. Data sets</a></li>
               <li><a href="Function_Glossary.html">A2. Function glossary</a></li>
            </ul>
          </li>
          
        <ul class="nav">
          <li><a href="Authors.html">About</a></li>
          <li class="dropdown">

        </ul>
      </div><!--/.nav-collapse -->
    </div>
  </div>
</div>

<div class="container">
<div class="row">
<div class="span10 offset1">

<div id="header">
<h1 class="title">A2. Function glossary</h1>
</div>

<div id="TOC">
<ul>
<li><a href="#navigation-in-r">Navigation in R</a></li>
<li><a href="#poppr-functions"><em>Poppr</em> functions</a></li>
</ul>
</div>

<p>To quit R you can either use the Rstudio&gt;Quit Rstudio pulldown menu command or execute ⌘+Q (OS X) or CTRL+Q (PC).</p>
<div id="navigation-in-r" class="section level2">
<h2>Navigation in R</h2>
<pre class="r"><code>getwd()        # this command will print the current working directory
ls()           # list objects in the current workspace
help(options)  # learn about available options
history()      # viewing your command history
data(*dsname*) # access builtin database *dsname*; must have pacakge loaded containing it
q()            # quit your R session

# default for saving history is &quot;.Rhistory&quot;
savehistory(file=&quot;myfile&quot;)  # save your command history for whole session
loadhistory(file= &quot;myfile&quot;) # recall command history save

# Package specific information
installed.packages()[,c(&quot;Package&quot;,&quot;Version&quot;)] #list packages &amp; versions installed
browseVignettes()                     # see vignettes from all packages

#Help on functions
help(functionname)     # Display documentation for a function
args(functionname)     # What are the arguments for a given function
example(functionname)  # see examples of function calls</code></pre>
</div>
<div id="poppr-functions" class="section level2">
<h2><em>Poppr</em> functions</h2>
<p>Our primer is heavily based on the <em>poppr</em> and <em>adegenet</em> packages. To get help on any of their functions type a question mark before the empty function call as in:</p>
<pre class="r"><code># Information on pacakges or functions
browseVignettes(package = &#39;poppr&#39;)    # see vignettes for poppr
vignette(&#39;poppr_manual&#39;)              # see manual for poppr
?mlg()           # open the R documentiaon of the function mlg()

# Poppr functions
help(package=&quot;poppr&quot;)   # shows all functions contained in poppr package</code></pre>
<div id="importexport" class="section level3">
<h3>Import/Export</h3>
<table>
<thead>
<tr class="header">
<th align="left">Function</th>
<th align="left">Description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left"><code>getfile</code></td>
<td align="left">Provides a quick GUI to grab files for import</td>
</tr>
<tr class="even">
<td align="left"><code>read.genalex</code></td>
<td align="left">Read <em>GenAlEx</em> formatted csv files to a genind object</td>
</tr>
<tr class="odd">
<td align="left"><code>genind2genalex</code></td>
<td align="left">Converts genind objects to <em>GenAlEx</em> formatted csv files</td>
</tr>
</tbody>
</table>
</div>
<div id="manipulation" class="section level3">
<h3>Manipulation</h3>
<table>
<thead>
<tr class="header">
<th align="left">Function</th>
<th align="left">Description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left"><code>as.genclone</code></td>
<td align="left">Converts genind objects to genclone objects</td>
</tr>
<tr class="even">
<td align="left"><code>setpop</code></td>
<td align="left">Set the population using defined hierarchies</td>
</tr>
<tr class="odd">
<td align="left"><code>splithierarchy</code></td>
<td align="left">Split a concatenated hierarchy imported as a population</td>
</tr>
<tr class="even">
<td align="left"><code>sethierarchy</code></td>
<td align="left">Define a population hierarchy of a genclone object</td>
</tr>
<tr class="odd">
<td align="left"><code>gethierarchy</code></td>
<td align="left">Extract the hierarchy data frame</td>
</tr>
<tr class="even">
<td align="left"><code>addhierarchy</code></td>
<td align="left">Add a vector or data frame to an existing hierarchy</td>
</tr>
<tr class="odd">
<td align="left"><code>namehierarchy</code></td>
<td align="left">Rename a population hierarchy</td>
</tr>
<tr class="even">
<td align="left"><code>missingno</code></td>
<td align="left">Handles missing data</td>
</tr>
<tr class="odd">
<td align="left"><code>clonecorrect</code></td>
<td align="left">Clone censors at a specified population hierarchy</td>
</tr>
<tr class="even">
<td align="left"><code>informloci</code></td>
<td align="left">Detects and removes phylogenetically uninformative loci</td>
</tr>
<tr class="odd">
<td align="left"><code>popsub</code></td>
<td align="left">Subsets genind objects by population</td>
</tr>
<tr class="even">
<td align="left"><code>shufflepop</code></td>
<td align="left">Shuffles genotypes at each locus using four different shuffling algorithms</td>
</tr>
<tr class="odd">
<td align="left"><code>recode_polyploids</code></td>
<td align="left">Recodes an imported polyploid data set to observed allelic frequencies</td>
</tr>
</tbody>
</table>
</div>
<div id="distances" class="section level3">
<h3>Distances</h3>
<table>
<thead>
<tr class="header">
<th align="left">Function</th>
<th align="left">Description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left"><code>bruvo.dist</code></td>
<td align="left">Bruvo’s distance</td>
</tr>
<tr class="even">
<td align="left"><code>diss.dist</code></td>
<td align="left">Absolute genetic distance (see <code>provesti.dist</code>)</td>
</tr>
<tr class="odd">
<td align="left"><code>nei.dist</code></td>
<td align="left">Nei’s 1978 genetic distance</td>
</tr>
<tr class="even">
<td align="left"><code>rogers.dist</code></td>
<td align="left">Rogers’ euclidean distance</td>
</tr>
<tr class="odd">
<td align="left"><code>reynolds.dist</code></td>
<td align="left">Reynolds’ coancestry distance</td>
</tr>
<tr class="even">
<td align="left"><code>edwards.dist</code></td>
<td align="left">Edwards’ angular distance</td>
</tr>
<tr class="odd">
<td align="left"><code>provesti.dist</code></td>
<td align="left">Provesti’s absolute genetic distance</td>
</tr>
</tbody>
</table>
</div>
<div id="bootstrapping" class="section level3">
<h3>Bootstrapping</h3>
<table>
<thead>
<tr class="header">
<th align="left">Function</th>
<th align="left">Description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left"><code>aboot</code></td>
<td align="left">Creates a bootstrapped dendrogram for any distance measure</td>
</tr>
<tr class="even">
<td align="left"><code>bruvo.boot</code></td>
<td align="left">Produces dendrograms with bootstrap support based on Bruvo’s distance</td>
</tr>
</tbody>
</table>
</div>
<div id="analysis" class="section level3">
<h3>Analysis</h3>
<table>
<thead>
<tr class="header">
<th align="left">Function</th>
<th align="left">Description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left"><code>poppr.amova</code></td>
<td align="left">Analysis of Molecular Variance (as implemented in ade4)</td>
</tr>
<tr class="even">
<td align="left"><code>ia</code></td>
<td align="left">Calculates the index of association</td>
</tr>
<tr class="odd">
<td align="left"><code>mlg</code></td>
<td align="left">Calculates the number of multilocus genotypes</td>
</tr>
<tr class="even">
<td align="left"><code>mlg.crosspop</code></td>
<td align="left">Finds all multilocus genotypes that cross populations</td>
</tr>
<tr class="odd">
<td align="left"><code>mlg.table</code></td>
<td align="left">Returns a table of populations by multilocus genotypes</td>
</tr>
<tr class="even">
<td align="left"><code>mlg.vector</code></td>
<td align="left">Returns a vector of a numeric multilocus genotype assignment for each individual</td>
</tr>
<tr class="odd">
<td align="left"><code>mlg.id</code></td>
<td align="left">Finds all individuals associated with each multilocus genotype</td>
</tr>
<tr class="even">
<td align="left"><code>poppr</code></td>
<td align="left">Returns a diversity table by population</td>
</tr>
<tr class="odd">
<td align="left"><code>poppr.all</code></td>
<td align="left">Returns a diversity table by population for all compatible files specified</td>
</tr>
<tr class="even">
<td align="left"><code>private_alleles</code></td>
<td align="left">Tabulates the occurences of alleles that only occur in one population</td>
</tr>
<tr class="odd">
<td align="left"><code>locus_table</code></td>
<td align="left">Create a table of summary statistics per locus</td>
</tr>
</tbody>
</table>
</div>
<div id="visualization" class="section level3">
<h3>Visualization</h3>
<table>
<thead>
<tr class="header">
<th align="left">Function</th>
<th align="left">Description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left"><code>plot_poppr_msn</code></td>
<td align="left">Plots minimum spanning networks produced in poppr with scale bar and legend</td>
</tr>
<tr class="even">
<td align="left"><code>greycurve</code></td>
<td align="left">Helper to determine the appropriate parameters for adjusting the grey level for msn functions</td>
</tr>
<tr class="odd">
<td align="left"><code>bruvo.msn</code></td>
<td align="left">Produces minimum spanning networks based off Bruvo’s distance colored by population</td>
</tr>
<tr class="even">
<td align="left"><code>poppr.msn</code></td>
<td align="left">Produces a minimum spanning network for any pairwise distance matrix related to the data</td>
</tr>
<tr class="odd">
<td align="left"><code>info_table</code></td>
<td align="left">Creates a heatmap representing missing data or observed ploidy</td>
</tr>
<tr class="even">
<td align="left"><code>genotype_curve</code></td>
<td align="left">Creates a series of boxplots to demonstrate how many markers are needed to represent the diversity of your data</td>
</tr>
</tbody>
</table>
<div class="references">

</div>
</div>
</div>

</div>
</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
$(document).ready(function () {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
});

</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
