<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />



<title>99: Minimum Spanning Networks</title>

<script src="libs/jquery-1.11.0/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="libs/bootstrap-3.3.1/css/readable.min.css" rel="stylesheet" />
<script src="libs/bootstrap-3.3.1/js/bootstrap.min.js"></script>
<script src="libs/bootstrap-3.3.1/shim/html5shiv.min.js"></script>
<script src="libs/bootstrap-3.3.1/shim/respond.min.js"></script>
<script>
  (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
  (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
  m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
  })(window,document,'script','//www.google-analytics.com/analytics.js','ga');

  ga('create', 'UA-52691601-2', 'auto');
  ga('send', 'pageview');

</script>

<!---

<style type="text/css">

/* padding for bootstrap navbar */
body {
  padding-top: 50px;
  padding-bottom: 40px;
  padding-right: 50px;
  padding-left: -50px;
}
@media (max-width: 979px) {
  body {
    padding-top: 0;
  }
}

 offset scroll position for anchor links (for fixed navbar)  
@media (min-width: 980px) {
  .section h2 {
    padding-top: 52px;
    margin-top: -52px;
  }
  .section h3 {
    padding-top: 52px;
    margin-top: -52px;
  }
}


/* don't use link color in navbar */
.dropdown-menu>li>a {
  color: black;
}

</style>
-->



<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 50px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h2 {
  padding-top: 55px;
  margin-top: -55px;
}
.section h3 {
  padding-top: 55px;
  margin-top: -55px;
}
/* don't use link color in navbar */
.dropdown-menu>li>a {
  color: black;
}
/* some padding for disqus */
#disqus_thread {
  margin-top: 45px;
}
</style>

<link rel="stylesheet" href="libs/font-awesome-4.1.0/css/font-awesome.min.css"/>


<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; background-color: #f8f8f8; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
pre, code { background-color: #f8f8f8; }
code > span.kw { color: #204a87; font-weight: bold; }
code > span.dt { color: #204a87; }
code > span.dv { color: #0000cf; }
code > span.bn { color: #0000cf; }
code > span.fl { color: #0000cf; }
code > span.ch { color: #4e9a06; }
code > span.st { color: #4e9a06; }
code > span.co { color: #8f5902; font-style: italic; }
code > span.ot { color: #8f5902; }
code > span.al { color: #ef2929; }
code > span.fu { color: #000000; }
code > span.er { font-weight: bold; }
</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>



</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img { 
  max-width:100%; 
  height: auto; 
}
</style>
<div class="container-fluid main-container">


<div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <!-- HOME -->
      <a class="navbar-brand" href="index.html"><b>Population Genetics in R</b></a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <!-- Table of contents -->
        <li class="dropdown">
          <!---<a href="chapters" class="dropdown-toggle" data-toggle="dropdown">Chapters<b class="caret"></b></a>-->
          <a href="chapters" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">Chapters <span class="caret"></span></a>
            <ol class="dropdown-menu">
               <li><a href="Preface.html">i. Preface</a></li>
               <li><a href="Introduction.html">1. Introduction</a></li>
               <li><a href="Getting_ready_to_use_R.html">2. Getting ready to use R</a></li>
               <li><a href="Data_Preparation.html">3. Data preparation</a></li>
               <li><a href="First_Steps.html">4. First steps</a></li>
               <li><a href="Population_Strata.html">5. Population strata and clone correction</a></li>
               <li><a href="Locus_Stats.html">6. Locus-based statistics and missing data</a></li>
               <li><a href="Genotypic_EvenRichDiv.html">7. Genotypic evenness, richness, and diversity</a></li>
               <li><a href="Linkage_disequilibrium.html">8. Linkage disequilibrium</a></li>
               <li><a href="Pop_Structure.html">9. Population structure</a></li>
               <li><a href="AMOVA.html">10. AMOVA</a></li>
               <li><a href="DAPC.html">11. Discriminant analysis of principal components (DAPC)</a></li>
            </ol>
        </li>
        <!-- Appendices dropdown -->
        <li class="dropdown">
          <!---<a href="appendices" class="dropdown-toggle" data-toggle="dropdown">Appendices<b class="caret"></b></a>-->
          <a href="appendices" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">Appendices <span class="caret"></span></a>
          <ul class = "dropdown-menu", role = "menu">
            <li><a href="Data_sets.html">A1. Data sets</a></li>
            <li><a href="Function_Glossary.html">A2. Function glossary</a></li>
            <li><a href="Intro_to_R.html">A3. Introduction to R</a></li>
          </ul>
        </li>
        <!-- Author button -->
        <li class="dropdown">
        	<a href="about" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">About<span class="caret"></span></a>
          <ul class = "dropdown-menu", role = "menu">
        	<li><a href="Authors.html">Authors</a></li>
        	<li>
        		<a href="https://github.com/grunwaldlab/Population_Genetics_in_R", target = "blank">
		    	Source code
		    	</a>
		    </li>
		   </ul>
        </li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->
<!-- 

old rmarkdown version: 

devtools::install_github("rstudio/rmarkdown", ref = "b96214b9ac86b437067a0aa21442203f52face83")


 -->
<!-- <div class="container">
<div class="row">
<div class="span10 offset1"> -->

<div id="header">
<h1 class="title">99: Minimum Spanning Networks</h1>
</div>

<div id="TOC">
<ul>
<li><a href="#from-the-command-line">From the command line</a></li>
<li><a href="#interactive-viewer">Interactive viewer</a></li>
</ul>
</div>

<p>Minimum spanning networks are a great way to be able to visualize the connections between samples in your data set. In terms of clonal organisms, it can be a more powerful visualization tool than trees, especially for clonal organisms. In this chapter, we will show you how to construct and view minimum spanning networks on the command line and in an interactive viewer.</p>
<div id="from-the-command-line" class="section level1">
<h1>From the command line</h1>
<div id="note" class="section level3">
<h3>Note:</h3>
<p>This section will utilize Bruvo’s distance <span class="citation">(Bruvo et al., 2004)</span>. It is important to note that calculating Bruvo’s distance is computationally different than other distances, so it has its own specialized functions for minimum spanning networks and bootstrapping that do not need a distance matrix:</p>
<table>
<thead>
<tr class="header">
<th align="left">General function</th>
<th align="left">Bruvo specific</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left"><code>poppr.msn</code></td>
<td align="left"><code>bruvo.msn</code></td>
</tr>
<tr class="even">
<td align="left"><code>aboot</code></td>
<td align="left"><code>bruvo.boot</code></td>
</tr>
</tbody>
</table>
</div>
<div id="minimum-spanning-network" class="section level3">
<h3>Minimum spanning network</h3>
<p>For this section, we will use the <code>monpop</code> data set from <span class="citation">(Everhart &amp; Scherm, 2015)</span>. See <a href="Genotypic_EvenRichDiv.html">Chapter 6</a> for more details. We will be focusing on sources of multilocus genotypes. The life cycle of the pathogen <em>Monilinia fructicola</em> begins with an ascospore (sexual propagule) released from a mummified peach fruit that had overwintered on the ground. It infects an emerging blossom that, in turn, asexually infects fruit, which proceed with cyclical infections. We might have questions such as:</p>
<ol style="list-style-type: decimal">
<li>Are the major genotypes of Fruit Rot (FR) samples closely related?</li>
<li>To what degree do the Blossom Blight (BB) samples contribute to the FR?</li>
</ol>
<p>Let’s load the data:</p>
<pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(<span class="st">&quot;poppr&quot;</span>)
<span class="kw">library</span>(<span class="st">&quot;magrittr&quot;</span>)
<span class="kw">data</span>(monpop)
<span class="kw">splitStrata</span>(monpop) &lt;-<span class="st"> </span><span class="er">~</span>Tree/Year/Symptom
<span class="kw">summary</span>(monpop)</code></pre>
<pre><code>## 
##  # Total number of genotypes:  694 
## 
##  # Population sample sizes:  
##  7_09_BB 26_09_BB 26_09_FR  7_09_FR 26_10_BB 45_10_BB 79_10_BB 79_10_FR 
##       23       41      132       73        5       13        1       64 
## 26_10_FR 45_10_FR 26_11_BB 26_11_FR 
##       85      130       30       97 
## 
##  # Number of alleles per locus:  
##  CHMFc4  CHMFc5 CHMFc12     SEA     SED     SEE     SEG     SEI     SEL 
##       3       6       3      11       9       5       8       7       5 
##     SEN     SEP     SEQ     SER 
##       9       8      11      10 
## 
##  # Number of alleles per population:  
##  7_09_BB 26_09_BB 26_09_FR  7_09_FR 26_10_BB 45_10_BB 79_10_BB 79_10_FR 
##       48       53       58       48       37       40       13       44 
## 26_10_FR 45_10_FR 26_11_BB 26_11_FR 
##       60       64       60       63 
## 
##  # Percentage of missing data:  
## [1] 0.5098648
## 
##  # Observed heterozygosity:  
## [1] 0
## 
##  # Expected heterozygosity:  
## [1] 0</code></pre>
<p>We notice that tree number 26 is the only one to have been sampled for all three years. Let’s use it as an example.</p>
<pre class="sourceCode r"><code class="sourceCode r">t26 &lt;-<span class="st"> </span>monpop %&gt;%<span class="st"> </span><span class="kw">setPop</span>(~Tree) %&gt;%<span class="st"> </span><span class="kw">popsub</span>(<span class="st">&quot;26&quot;</span>) %&gt;%<span class="st"> </span><span class="kw">setPop</span>(~Year/Symptom)
t26</code></pre>
<pre><code>## 
## This is a genclone object
## -------------------------
## Genotype information:
## 
##    155 multilocus genotypes 
##    390 haploid individuals
##     13 codominant loci
## 
## Population information:
## 
##      3 strata - Tree Year Symptom
##      6 populations defined - 9_BB 9_FR 10_BB 10_FR 11_BB 11_FR</code></pre>
<p>Now that we have our tree, let’s calculate a MSN using Bruvo’s distance <span class="citation">(Bruvo et al., 2004)</span>. Remember that this distance is based on a stepwise mutation model, so we have to first specify what kind of repeats units we have in our data (eg. dinucleotide = 2, trinucleotide = 3, etc.):</p>
<pre class="sourceCode r"><code class="sourceCode r"><span class="co"># Set up our repeat lengths and populations to analyze</span>
reps &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="dt">CHMFc4 =</span> <span class="dv">7</span>, <span class="dt">CHMFc5 =</span> <span class="dv">2</span>, <span class="dt">CHMFc12 =</span> <span class="dv">4</span>, 
          <span class="dt">SEA =</span> <span class="dv">4</span>, <span class="dt">SED =</span> <span class="dv">4</span>, <span class="dt">SEE =</span> <span class="dv">2</span>, <span class="dt">SEG =</span> <span class="dv">6</span>,
          <span class="dt">SEI =</span> <span class="dv">3</span>, <span class="dt">SEL =</span> <span class="dv">4</span>, <span class="dt">SEN =</span> <span class="dv">2</span>, 
          <span class="dt">SEP =</span> <span class="dv">4</span>, <span class="dt">SEQ =</span> <span class="dv">2</span>, <span class="dt">SER =</span> <span class="dv">4</span>)

sub9 &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;9_BB&quot;</span>, <span class="st">&quot;9_FR&quot;</span>)

<span class="co"># Calculate the MSN</span>
t26.9msn &lt;-<span class="st"> </span><span class="kw">bruvo.msn</span>(t26, <span class="dt">replen =</span> reps, <span class="dt">sublist =</span> sub9, <span class="dt">showplot =</span> <span class="ot">FALSE</span>)</code></pre>
<p>The minimum spanning network is calculated via <code>bruvo.msn</code>. I’ve set the argument <code>showplot = FALSE</code> because I want to use the more powerful function <code>plot_poppr_msn</code> to view the MSN. In this, I’m telling it to lable none of the samples, color populations using the “cm.colors” palette and scale the size of the nodes to <span class="math">\(log_{1.25}\)</span>. If you want to know what other things this function can do, simply type <code>help(&quot;plot_poppr_msn&quot;)</code></p>
<pre class="sourceCode r"><code class="sourceCode r"><span class="co"># Visualize the network</span>
<span class="kw">set.seed</span>(<span class="dv">120</span>)
<span class="kw">plot_poppr_msn</span>(t26, t26.9msn, <span class="dt">inds =</span> <span class="st">&quot;none&quot;</span>, <span class="dt">palette =</span> cm.colors, <span class="dt">nodebase =</span> <span class="fl">1.25</span>)</code></pre>
<p><img src="Minimum_Spanning_Networks_files/figure-html/visualize_msn-1.png" title="" alt="" width="700px" /></p>
<p>We can see that the Blossom Blight in the tree (pink pie pieces) heavily contributed to the major groups of MLGs found in the Fruit Rot (blue pie pieces).</p>
<blockquote>
<p><strong>Try it!</strong> See if you can produce similar graphs with the 2010 and 2011 populations.</p>
</blockquote>
</div>
</div>
<div id="interactive-viewer" class="section level1">
<h1>Interactive viewer</h1>
<p>That seems like a lot of work to create a minimum spanning network, right? If you look at the all of the arguments for <code>plot\_poppr\_msn</code>, there are a whole lot of them:</p>
<pre class="sourceCode r"><code class="sourceCode r"><span class="kw">args</span>(plot_poppr_msn)</code></pre>
<pre><code>## function (x, poppr_msn, gscale = TRUE, gadj = 3, mlg.compute = &quot;original&quot;, 
##     glim = c(0, 0.8), gweight = 1, wscale = TRUE, nodebase = 1.15, 
##     nodelab = 2, inds = &quot;ALL&quot;, mlg = FALSE, quantiles = TRUE, 
##     cutoff = NULL, palette = NULL, layfun = layout.auto, beforecut = FALSE, 
##     pop.leg = TRUE, scale.leg = TRUE, ...) 
## NULL</code></pre>
<p>This is overwhelming even for people who are comfortable with R. Since this function is designed to provide a visualization after calculation, it only makes sense to have an interactive, visual interface. This interface is called <code>imsn()</code>. Let’s try to recreate the plot above using the <code>imsn()</code> function. Note that the function takes no arguments.</p>
<pre class="sourceCode r"><code class="sourceCode r"><span class="kw">imsn</span>()</code></pre>
<p>(Note: you probably will not see the whole screen like this, but you can still scroll down)</p>
<p><img src="images/imsn_initial.png" alt="Intial view of imsn" /></p>
<p>We can see that there are five tabs and a sidebar. For now, we’ll explain what is on the sidebar. The first thing you see is a green bar that says “ready”. This means that there are no processes going on in the background and that it’s ready for input. Below that are three buttons, <strong>Go!</strong>, <strong>reData</strong>, and <strong>reGraph</strong>. These buttons are how you will tell the program that you are ready for it to make your graph.</p>
<p>Notice that there are two main sections, <strong>Data Parameters</strong> and <strong>Graphical Parameters</strong>.</p>
<div id="data-parameters" class="section level3">
<h3>Data Parameters</h3>
<p>As you can expect, these options represent parameters that will manipulate your data. Changing any of these choices means that the distance matrix will be recalculated. You see here four choices presented:</p>
<ol style="list-style-type: decimal">
<li>choose dataset - This dropdown menu allows you to choose your data set.<br /><img src="images/imsn_data_choose.png" alt="choose dataset" /></li>
<li>choose populations - this will change based on your data set, but it allows you to select different populations with which to subset your data<br /><img src="images/imsn_population_choose.png" alt="choose populations" /></li>
<li>convert to genclone? - If your data is in genind format, you can convert it to genclone (minimally reduces time for clonal organisms)<br /><img src="images/imsn_convert.png" alt="convert to genclone?" /></li>
<li>choose distance calculation - The distance measure to calculate from your data (notice that there’s “Custom” at the bottom)<br /><img src="images/imsn_distance_choose.png" alt="choose distance calculation" /></li>
<li>Distance arguments - every distance has different arguments, this is where you can modify them<br /><img src="images/imsn_distance_args.png" alt="Distance arguments" /></li>
<li>include reticulations? - This is an important parameter that will include reticulations in the minimum spanning network. We’ll explore this with the <code>Pram</code> data set later.<br /><img src="images/imsn_reticulate.png" alt="include reticulations?" /></li>
</ol>
<div class="references">
<p>Bruvo R., Michiels NK., D’Souza TG., Schulenburg H. 2004. A simple method for the calculation of microsatellite genotype distances irrespective of ploidy level. <em>Molecular Ecology</em> 13:2101–2106. Available at: <a href="http://dx.doi.org/10.1111/j.1365-294X.2004.02209.x" class="uri">http://dx.doi.org/10.1111/j.1365-294X.2004.02209.x</a></p>
<p>Everhart S., Scherm H. 2015. Fine-scale genetic structure of <em>Monilinia fructicola</em> during brown rot epidemics within individual peach tree canopies. <em>Phytopathology</em> 105:542–549.</p>
</div>
</div>
</div>

<!-- </div>
</div>
</div>
 -->

</div>

<script>

// add bootstrap table styles to pandoc tables
$(document).ready(function () {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
});

</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
